<!DOCTYPE html>



  


<html class="theme-next pisces use-motion" lang="zh-Hans">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>
<meta name="theme-color" content="#222">



  
  
    
    
  <script src="/lib/pace/pace.min.js?v=1.0.2"></script>
  <link href="/lib/pace/pace-theme-minimal.min.css?v=1.0.2" rel="stylesheet">
  <script>
    (function(){
      if(''){
        if (prompt('请输入文章密码') !== ''){
          alert('密码错误！');
          history.back();
        }
      }
    })();
  </script>







<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css" />







<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css" />

<link href="/css/main.css?v=5.1.2" rel="stylesheet" type="text/css" />


  <meta name="keywords" content="Standards and guidelines," />








  <link rel="shortcut icon" type="image/x-icon" href="/images/favicon.ico?v=5.1.2" />






<meta name="description" content="肿瘤体系系变异解读指南">
<meta name="keywords" content="Standards and guidelines">
<meta property="og:type" content="article">
<meta property="og:title" content="Standards and Guidelines for the Interpretation and Reporting of Sequence Variants in Cancer">
<meta property="og:url" content="http://ben-air.cn/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/index.html">
<meta property="og:site_name" content="Ben-air">
<meta property="og:description" content="肿瘤体系系变异解读指南">
<meta property="og:locale" content="zh-Hans">
<meta property="og:image" content="http://ben-air.cn/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/Working_Group_technical_and_reporting_survey.jpg">
<meta property="og:image" content="http://ben-air.cn/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/与体细胞序列变异解释有关的数据库.jpg">
<meta property="og:image" content="http://ben-air.cn/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/Algorithms_for_Computational_Prediction_of_Functional_Impact_of_Sequence_VariantSplice_Site_Changes.jpg">
<meta property="og:image" content="http://ben-air.cn/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/Categories_of_Clinical_andor_Experimental_Evidence.jpg">
<meta property="og:image" content="http://ben-air.cn/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/体细胞变异等级和Level关系.jpg">
<meta property="og:image" content="http://ben-air.cn/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/Tier_I.jpg">
<meta property="og:image" content="http://ben-air.cn/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/Tier_II.jpg">
<meta property="og:image" content="http://ben-air.cn/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/Tier_III.jpg">
<meta property="og:image" content="http://ben-air.cn/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/Tier_IV.jpg">
<meta property="og:updated_time" content="2021-12-28T06:19:14.050Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Standards and Guidelines for the Interpretation and Reporting of Sequence Variants in Cancer">
<meta name="twitter:description" content="肿瘤体系系变异解读指南">
<meta name="twitter:image" content="http://ben-air.cn/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/Working_Group_technical_and_reporting_survey.jpg">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Pisces',
    sidebar: {"position":"left","display":"post","offset":12,"offset_float":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="http://ben-air.cn/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/"/>





  <title>Standards and Guidelines for the Interpretation and Reporting of Sequence Variants in Cancer | Ben-air</title>
  














</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  
  
    
  

  <div class="container sidebar-position-left page-post-detail ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <div class="custom-logo-site-title">
      <a href="/"  class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">Ben-air</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle"></p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br />
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br />
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br />
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br />
            
            标签
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br />
            
            关于
          </a>
        </li>
      

      
        <li class="menu-item menu-item-search">
          
            <a href="javascript:;" class="popup-trigger">
          
            
              <i class="menu-item-icon fa fa-search fa-fw"></i> <br />
            
            搜索
          </a>
        </li>
      
    </ul>
  

  
    <div class="site-search">
      
  <div class="popup search-popup local-search-popup">
  <div class="local-search-header clearfix">
    <span class="search-icon">
      <i class="fa fa-search"></i>
    </span>
    <span class="popup-btn-close">
      <i class="fa fa-times-circle"></i>
    </span>
    <div class="local-search-input-wrapper">
      <input autocomplete="off"
             placeholder="搜索..." spellcheck="false"
             type="text" id="local-search-input">
    </div>
  </div>
  <div id="local-search-result"></div>
</div>



    </div>
  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://ben-air.cn/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="Ben-air">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="/uploads/avatar.jpg">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Ben-air">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">Standards and Guidelines for the Interpretation and Reporting of Sequence Variants in Cancer</h1>
        

        <div class="post-meta">
	  

          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2021-02-19T17:53:48+08:00">
                2021-02-19
              </time>
            
            

            

            
          </span>

          
            <span class="post-category" >
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">分类于</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/NGS/" itemprop="url" rel="index">
                    <span itemprop="name">NGS</span>
                  </a>
                </span>

                
                
                  ， 
                
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/NGS/标准化/" itemprop="url" rel="index">
                    <span itemprop="name">标准化</span>
                  </a>
                </span>

                
                
                  ， 
                
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/NGS/肿瘤检测/" itemprop="url" rel="index">
                    <span itemprop="name">肿瘤检测</span>
                  </a>
                </span>

                
                
              
            </span>
          

          
            
          

          
          

          
            <span class="post-meta-divider">|</span>
            <span class="page-pv"><i class="fa fa-file-o"></i>
            <span class="busuanzi-value" id="busuanzi_value_page_pv" ></span>
            </span>
          

          

          
              <div class="post-description">
                  肿瘤体系系变异解读指南
              </div>
          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">
      
      

      
        <h3 id="参考文献"><a href="#参考文献" class="headerlink" title="参考文献"></a>参考文献</h3><p><a href="Standards_and_Guidelines_for_the_Interpretation_and_Reporting_of_Sequence_Variants_in_Cancer.pdf">Standards and Guidelines for the Interpretation and Reporting of Sequence Variants in Cancer</a></p>
<p><a href="癌症变异解读和报告指南.pdf">整理汇总PPT材料</a></p>
<h3 id="摘要"><a href="#摘要" class="headerlink" title="摘要"></a>摘要</h3><p>下一代基于测序的癌症测试的临床实验室的广泛实施突出了标准化实验室间分子结果的解释和报告的重要性和潜在的好处。分子病理学协会召集了一个多学科工作组，负责评估基于下一代测序技术的癌症检测的现状，并建立体细胞序列变异的标准化共识分类，注释，解释和报告惯例，并由美国医学遗传学和基因组学学院联络代表，美国临床肿瘤学会和美国病理学家学院。根据专业调查，文献综述和工作组主题专家共识的结果，提出了一种基于体细胞序列变异的临床意义进行分类的四级系统：一级，具有很强临床意义的变异； II级，具有潜在临床意义的变体； III级，临床意义未知的变体；和第四级，变体被视为良性或可能良性。癌症基因组学是一个快速发展的领域。因此，应不断评估治疗，诊断或预后中任何变异的临床意义。报告基因组变异应遵循标准命名法，并明确描述测试方法和限制。临床建议应简明扼要，并与组织学和临床发现相关联。 </p>
<h4 id="分类"><a href="#分类" class="headerlink" title="分类"></a>分类</h4><ul>
<li>tier I, variants with strong clinical significance;</li>
<li>tier II, variants with potential clinical significance;</li>
<li>tier III, variants of unknown clinical significance;</li>
<li>tier IV, variants deemed benign or likely benign.</li>
</ul>
<h3 id="导读"><a href="#导读" class="headerlink" title="导读"></a>导读</h3><p>基于二代测序技术的肿瘤检测已越来越多的应用于临床实验室中，但目前在不同实验室间存在检测方法、报告内容等方面的差异，这对遗传检测的解读以及普及应用造成了一定的影响。因此，在不同实验室间建立统一的分子检测结果的解读和报告标准，及建立行业标准，显得尤为迫切。</p>
<p>在2015年春天，在美国专门成立了一个以临床实验室为核心的工作组，其组成包含了分子病理协会（AMP）、美国医学遗传学与基因组学协会（ACMGG）、美国临床肿瘤学会(ASCO）与美国病理学家协会（CAP)的一线专家，该工作组的主要工作为对肿瘤及疑似肿瘤相关的序列变异检测建立检测标准并在行业达成共识。</p>
<p>该工作组首先对北美地区超过40家的临床检测实验室进行了问卷调查，结果显示不同实验室在检测组织类型、检测基因数量、是否检测肿瘤组织全外显子组或全基因组、以及其他细节方面都存在较大差别，此外在不同单位的检测报告的报告内容方面也存在较多差异。该工作组认为，为医疗机构提供分类的遗传变异报告对病人及整个医疗行业都极为重要，包括：提供精确的肿瘤对靶向治疗反应性信息；建立国家级别的医疗指南；以及与临床试验合作，对建立不同实验室间的通用标准提供支持。基于以上这些考量，工作组专家们根据已有数据、文献报道和专业知识，给出以下指南建议.</p>
<h4 id="对多个公司的报告情况调研结果如下"><a href="#对多个公司的报告情况调研结果如下" class="headerlink" title="对多个公司的报告情况调研结果如下"></a>对多个公司的报告情况调研结果如下</h4><p><img src="/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/Working_Group_technical_and_reporting_survey.jpg" alt="image"></p>
<pre><code>图1 AMP对NGS技术及NGS结果解读的调研 A: MAF阈值. B: 变异分类数目 C:报告中是否包含治疗性建议 . D: 报告中是否包含潜在的生殖细胞突变. E: 报告是否包含变异等位频率Variant allele frequency (VAF) F: 报告是否包含基因组坐标 G: 报告是否包含转录本ID(Transcript accession) H: 报告是否包含不符合质控的基因/区间
</code></pre><h3 id="数据库"><a href="#数据库" class="headerlink" title="数据库"></a>数据库</h3><p><img src="/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/与体细胞序列变异解释有关的数据库.jpg" alt="image"></p>
<h4 id="Genomic-Databases"><a href="#Genomic-Databases" class="headerlink" title="Genomic Databases"></a>Genomic Databases</h4><p>随着越来越多的针对各种肿瘤类型的大规模基因组测序项目的发布，全球正在产生大量的基因组信息并将其整合到许多公共数据库中（表1）。例如，美国国家癌症研究所（National Cancer Institute）的基因组数据共享（Genome Data Commons）包含美国国家癌症研究所（National Institutes Institutes）从一些最大和最全面的癌症基因组数据集中生成的数据.包括 ：</p>
<pre><code>The Cancer Genome Atlas, Therapeutically Applicable Research to Generate Effective Therapies, and the Cancer Genome Characterization Initiative (https://gdc.cancer.gov, last accessed September 25, 2016). Another public somatic variant database is the Catalog of Somatic Mutations in Cancer (http://cancer.sanger.ac.uk/cosmic, last accessed September 30, 2016).
</code></pre><p>其中包含数百万种跨多种肿瘤类型的体细胞变异。体细胞变异分析中经常使用的其他几个数据库，例如参考序列信息，种群数据库和种系变异数据库也在不断增加和改进。基因组数据库提供了准确注释和确定体变异优先级所必需的信息。</p>
<p>在使用这些数据库是，整体上应该遵循乳癌规则： </p>
<ol>
<li>了解数据库的内容以及如何汇总数据。临床实验室应查看与给定数据库有关的文档或公开文献，以确定数据库的来源，类型和意图</li>
<li>特别注意每个数据库的限制，以避免对注释结果的过度解释。</li>
<li>确认人类基因组装配的版本以及mRNA转录本参考，以确保适当的人类基因组变异学会（HGVS）注释。</li>
<li>尽可能使用基因组坐标而不是HGVS命名法来明确查询基因组数据库。</li>
<li>根据出版物或其他数据库的来源，单个或多个特定条目的数量，研究的深度，使用适当的对照，确认变异的体细胞来源，评估提供的基因组数据的质量以及功能和潜在药物反应研究。</li>
<li>验证所提供病理诊断的数据质量（例如，地点，诊断和子类型</li>
</ol>
<h4 id="Reference-Sequence-Databases"><a href="#Reference-Sequence-Databases" class="headerlink" title="Reference Sequence Databases"></a>Reference Sequence Databases</h4><p>参考序列数据库提供有关人类基因组装配版本的信息以及相关信息，例如基因组坐标，以明确表示序列变异。附加信息，例如mRNA转录本的登录和版本（例如，BRAF NM_004333.4）和外显子边界定义，对于产生变异的正确HGVS命名法至关重要。可以从这些数据库中计算基因的变异位置图谱（编码，非编码，非翻译区和剪接位点）和链表述（阳性与阴性）。这也允许在没有基因组坐标信息的情况下明确表示变异。一些常用资源包括：</p>
<ul>
<li>RefSeq（国家生物技术信息中心参考序列数据库，https：//<a href="http://www.ncbi.nlm.nih.gov/refseq，上次访问时间为2016年1月2日）" target="_blank" rel="noopener">www.ncbi.nlm.nih.gov/refseq，上次访问时间为2016年1月2日）</a></li>
<li>Ensembl（<a href="http://www.Ensembl.org）。" target="_blank" rel="noopener">http://www.Ensembl.org）。</a> ensembl.org/index.html，最后访问时间为2016年1月2日）</li>
<li>Locus Reference Genomic （<a href="https://www.lrg-sequence.org，最近访问时间为2016年2月2日）。" target="_blank" rel="noopener">https://www.lrg-sequence.org，最近访问时间为2016年2月2日）。</a> </li>
</ul>
<h4 id="Population-Databases"><a href="#Population-Databases" class="headerlink" title="Population Databases"></a>Population Databases</h4><p>这些数据库提供了有关大量特定人群中给定基因座上替代（次要）等位基因频率的全面信息。这些数据库通常用于根据次要等位基因频率（MAF）的任意临界值筛选出被认为是多态/良性的变异。目前尚无用于去除多态或良性变异的MAF的标准临界值。在没有正常组织配对的情况下，工作组建议使用<strong>1％（0.01）作为主要阈值</strong>，这在许多临床实验室中也很普遍。尽管总体MAF最常用，但临床实验室可能会考虑使用种族-根据患者的种族背景确定特定的MAF。在解释体细胞变异时，必须谨慎使用这些数据库，因为在参与研究时，假定参与这些测序研究的个体是健康的或没有亚临床疾病。确实，一些众所周知的经典癌症相关的和可靶向的体细胞改变已作为种群数据库的种系变异包括在内。例如，变异NM_004972.3（JAK2）：c.1849G&gt; T<br>（c.V617F）通常被看作是叶绿体增生的体细胞变异体<br>肿瘤，可以用FDA批准的Janus激酶（JAK）抑制剂靶向。它也包含在多个人群数据库中，例如 <strong>The Database of Short Genetic Variation (the National Center for Biotechnology Information database of genetic variation), Exome Variant Server, and Exome Aggregation Consortium</strong>（表格1）。<strong>在评估可能的血液系统恶性肿瘤时应格外小心，因为白血病和骨髓增生异常综合症中的许多常见突变基因也可能在其他健康个体的血液中发生体细胞突变，因此可能被错误地注释为多态性</strong>。</p>
<h4 id="Cancer-Specific-Databases"><a href="#Cancer-Specific-Databases" class="headerlink" title="Cancer-Specific Databases"></a>Cancer-Specific Databases</h4><p>这些数据库提供了有关不同癌症和亚型谱中序列变异的发生率和普遍性的信息，对其他基因组数据库的交叉引用以及对已发表或未进行系统综述的文献的引用，细胞途径，靶向疗法，临床试验 以及结果数据。 从这些数据库中提取的不同癌症中的序列变异体的普遍性和分布，应谨慎解释，因为病理诊断标准的代表性较差，缺乏临床级别的文献管理以及提交变异体的来源控制不严（例如，探索性或 发现研究）。 例如，这些数据库中包括一些常见的种系良性变异，例如 <strong>the Catalog of Somatic Mutations in Cancer database</strong> 中的NM_000222.2（KIT）：c.1621A&gt; C（p.M541L）。 表1列出了常用的体细胞变异数据库。 </p>
<h4 id="Constitutional-Variant-Databases"><a href="#Constitutional-Variant-Databases" class="headerlink" title="Constitutional Variant Databases"></a>Constitutional Variant Databases</h4><p>常见的是，在有或没有匹配的正常组织的情况下进行肿瘤测序可能会揭示出种系起源的变异，例如与癌症易感综合症相关的基因中的致病变异。种系突变数据库，例如人类基因突变数据库和其他疾病或基因座特异性突变数据库，是评估这些变异的有用资源。这些数据库也可用于评估在这些数据库中报道了经过充分研究的种系对应物的体细胞变异（例如，TP53和PTEN基因中的某些变异）。另一个常用的数据库是ClinVar（<a href="http://www.ncbi.nlm.nih.gov/clinvar）。" target="_blank" rel="noopener">http://www.ncbi.nlm.nih.gov/clinvar）。</a> ClinVar处理所有种类的稀有种系变异，例如病原体和良性，并在可用时提供相关的临床和实验证据。专家小组对ClinVar中的某些变异进行了有关其致病性的审查。目前，该数据库仅宿主种系变异，并有望在不久的将来纳入体细胞变异。</p>
<h4 id="Internal-Laboratory-Generated-Databases"><a href="#Internal-Laboratory-Generated-Databases" class="headerlink" title="Internal (Laboratory-Generated) Databases"></a>Internal (Laboratory-Generated) Databases</h4><p>需要强调的是，临床实验室应该建立一个标注良好的内部数据库，以跟踪实验室中识别出的变异并提供一致的变异注释。这样的数据库可用于识别可能由测序比对伪像引起的潜在假阳性呼叫，以及确定实验室通常遇到的癌症类型的突变频率。我们强烈鼓励体细胞变异数据共享，并敦促临床实验室将精心挑选的变异体贡献到公共变异数据库中，以促进对体细胞变异体的准确解释。但是，此类提交过程应标准化并符合联邦隐私法规，即《健康保险可移植性和责任法案》以及《经济和临床健康信息技术法案》（原文 the Health Insurance Portability and Accountability Act and<br>the Health Information Technology for Economic and Clinical<br>Health Act.）。 正在努力建立临床级基因组数据库 </p>
<h3 id="In-Silico-Computational-Prediction-Algorithms"><a href="#In-Silico-Computational-Prediction-Algorithms" class="headerlink" title="In Silico (Computational) Prediction Algorithms"></a>In Silico (Computational) Prediction Algorithms</h3><p><img src="/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/Algorithms_for_Computational_Prediction_of_Functional_Impact_of_Sequence_VariantSplice_Site_Changes.jpg" alt="image"></p>
<p>在计算机模拟中，预测算法是预测基因中核苷酸变化是否会改变蛋白质结构和功能的常用工具（表2）。起初，开发了早期常用算法并验证了用于胚系变异的算法。随后，外推它们的用途以解释体细胞变异。尽管各个算法的核心风险预测方法可能有所不同，但是它们可以分为两类：错义变异对蛋白质功能的影响的预测和序列变异对剪接的影响。考虑到氨基酸或核苷酸残基的进化保守程度，特定理化特性的氨基酸取代的生物化学影响以及变异在翻译蛋白质中的位置，是不同算法用来预测功能的一些主要标准错义变异的影响。拼接位点预测算法使用各种统计方法，例如马尔可夫模型，机器学习（神经网络）和最大熵原理，来预测变异是否会对拼接产生任何影响。通常，错义和剪接位点预测工具具有中等的特异性（大约60％至80％），并且倾向于过度预测有害影响。<br>在癌症基因功能的背景下，对这些预测的解释通常并不直接了当，特别是对于激活突变。例如，当通过多种计算机模拟算法进行分析时，经典的BRAF V600E致癌突变结果会具有冲突甚至良性的作用。这是临床实验室在进行体细胞变异解释时应该意识到的几种情况之一，同时在解释计算机分析的评分结果时应谨慎行事。建议不要将这些预测算法的结果用作变异分类或临床决策的唯一依据。</p>
<h3 id="Variant-Identification-and-Annotation"><a href="#Variant-Identification-and-Annotation" class="headerlink" title="Variant Identification and Annotation"></a>Variant Identification and Annotation</h3><p>变异检测是变异解释的关键起点。 有许多变异检测软件工具(Supplemental Table S2).可以满足一种特定的检测，例如SNV，插入缺失，结构变异和CNV </p>
<p><strong>(Supplemental Table S1)</strong>:</p>
<table>
<thead>
<tr>
<th style="text-align:center">Variant caller</th>
<th style="text-align:center">Location (URL)</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center">MuTect v1.1.555</td>
<td style="text-align:center"><a href="https://www.broadinstitute.org/cancer/cga/mutect" target="_blank" rel="noopener">https://www.broadinstitute.org/cancer/cga/mutect</a></td>
</tr>
<tr>
<td style="text-align:center">Genome Analysis Toolkit (GATK) – MuTect v2s</td>
<td style="text-align:center"><a href="https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.php" target="_blank" rel="noopener">https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.php</a></td>
</tr>
<tr>
<td style="text-align:center">VarScan 256</td>
<td style="text-align:center"><a href="http://dkoboldt.github.io/varscan/" target="_blank" rel="noopener">http://dkoboldt.github.io/varscan/</a></td>
</tr>
<tr>
<td style="text-align:center">VarDict57</td>
<td style="text-align:center"><a href="https://github.com/AstraZeneca-NGS/VarDict" target="_blank" rel="noopener">https://github.com/AstraZeneca-NGS/VarDict</a></td>
</tr>
<tr>
<td style="text-align:center">Sterlka58</td>
<td style="text-align:center"><a href="https://sites.google.com/site/strelkasomaticvariantcaller/" target="_blank" rel="noopener">https://sites.google.com/site/strelkasomaticvariantcaller/</a></td>
</tr>
<tr>
<td style="text-align:center">FreeBayes59</td>
<td style="text-align:center"><a href="https://github.com/ekg/freebayes" target="_blank" rel="noopener">https://github.com/ekg/freebayes</a></td>
</tr>
<tr>
<td style="text-align:center">Scalpel60</td>
<td style="text-align:center"><a href="http://scalpel.sourceforge.net/" target="_blank" rel="noopener">http://scalpel.sourceforge.net/</a></td>
</tr>
<tr>
<td style="text-align:center">Pindel61</td>
<td style="text-align:center"><a href="http://gmt.genome.wustl.edu/packages/pindel/" target="_blank" rel="noopener">http://gmt.genome.wustl.edu/packages/pindel/</a></td>
</tr>
<tr>
<td style="text-align:center">SAMtools62</td>
<td style="text-align:center"><a href="http://samtools.sourceforge.net/" target="_blank" rel="noopener">http://samtools.sourceforge.net/</a></td>
</tr>
<tr>
<td style="text-align:center">Torrent Suite Variant Caller</td>
<td style="text-align:center"><a href="https://github.com/iontorrent/TS" target="_blank" rel="noopener">https://github.com/iontorrent/TS</a></td>
</tr>
<tr>
<td style="text-align:center">SomaticSniper63</td>
<td style="text-align:center"><a href="http://gmt.genome.wustl.edu/packages/somatic-sniper/" target="_blank" rel="noopener">http://gmt.genome.wustl.edu/packages/somatic-sniper/</a></td>
</tr>
</tbody>
</table>
<p>对于临床实验室来说，了解这些变异检测工具的局限性很重要。 对生物信息学流程（包括商业购买的生物信息学软件包）进行适当的实验室验证对于确保结果的质量至关重要。 某些称为变异的指标对于变异解释至关重要，例如 <strong>supporting reads (depth of coverage) and variant allele frequency (VAF)</strong>，应纳入变异体评估中； 后者对于在没有配对正常的情况下的体细胞变异解释和评估肿瘤克隆多样性特别重要。 </p>
<p>变异检测结果一般使用一些标准格式进行输出展示，例如clinical variant call format(VCF), genomic VCF 和 general feature format (alias gene-finding format or generic feature format).</p>
<p>The VCF is the most widely used schema in the clinical laboratories as of 2016 to represent detected variants (Clinical Variant Call Format, <a href="http://vcfclin.org" target="_blank" rel="noopener">http://vcfclin.org</a>, last accessed September 28, 2016). Required VCF fields include genomic coordinates, reference nucleotide(s), and variant nucleotide( s). However, complex, multinucleotide, and large structural variants are difficult to represent in the current specification of VCF file format version 4.2, despite the ongoing efforts for standardizing variant representation.</p>
<p>For further clinical interpretation, additional metadata that add meaningful and readily identifiable information to variants should be included (eg, <strong>gene symbol</strong>, <strong>variant location</strong>, <strong>variant type</strong>, <strong>HGVS nomenclature for cDNA sequence changes</strong>, and <strong>predicted protein sequence alterations</strong>). Additional resources, such as cross-references to external databases (cancerspecific and general genomic databases) (Table 1) and precomputed in silico algorithm-based predictions (Table 2), can also be beneficial. This process is formally referred to as variant annotation, and may be automated by software tools(Supplemental Table S2).</p>
<p><strong>(Supplemental Table S2)</strong>:</p>
<table>
<thead>
<tr>
<th style="text-align:center">Software</th>
<th style="text-align:center">Location (URL)</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center">Annovar64</td>
<td style="text-align:center"><a href="http://annovar.openbioinformatics.org/en/latest/" target="_blank" rel="noopener">http://annovar.openbioinformatics.org/en/latest/</a></td>
</tr>
<tr>
<td style="text-align:center">snpEff65</td>
<td style="text-align:center"><a href="http://snpeff.sourceforge.net/" target="_blank" rel="noopener">http://snpeff.sourceforge.net/</a></td>
</tr>
<tr>
<td style="text-align:center">SeattleSeq</td>
<td style="text-align:center"><a href="http://snp.gs.washington.edu/SeattleSeqAnnotation138/" target="_blank" rel="noopener">http://snp.gs.washington.edu/SeattleSeqAnnotation138/</a></td>
</tr>
<tr>
<td style="text-align:center">AnnTools66</td>
<td style="text-align:center"><a href="http://anntools.sourceforge.net/" target="_blank" rel="noopener">http://anntools.sourceforge.net/</a></td>
</tr>
<tr>
<td style="text-align:center">NGS-SNP67</td>
<td style="text-align:center"><a href="https://www.ualberta.ca/~stothard/downloads/NGS-SNP/" target="_blank" rel="noopener">https://www.ualberta.ca/~stothard/downloads/NGS-SNP/</a></td>
</tr>
<tr>
<td style="text-align:center">VEP (Variant Effect Predictor)15</td>
<td style="text-align:center"><a href="http://useast.ensembl.org/info/docs/tools/vep/index.html" target="_blank" rel="noopener">http://useast.ensembl.org/info/docs/tools/vep/index.html</a></td>
</tr>
</tbody>
</table>
<p>变异注释对于准确解释体细胞序列变异至关重要。这些对变异注释得到的多元数据构成了变异评估和解析的初始内容。<br>变异注释面对的一个重要挑战就是将基因组坐标（即染色体和位置）转换为相应的cDNA /氨基酸坐标系统（分别为c.和p.syntax）以进行解读。</p>
<p>这个问题针对indel变异由于对齐排列导致变异表述不一致的问题上，表现尤其突出。尽管HGVS系统建议使用右对齐表示法（将变异的开始位置向右移动，直到不再可能这样做），但VCF规范要求使用左对齐表示法。当前可用的注释解决方案仅部分解决了该问题。缺乏左/右对齐的标准化可能会严重影响变异定位，从而导致变异命名错误。根据HGVS命名法，当存在多个替代转录本时，必须使用正确的mRNA转录本编号和版本信息，以确保变异描述的准确和一致。临床实验室在内部数据库中使用变异的基因组位置来存储变异信息也非常重要，以确保数据存储的明确和可回查。</p>
<h4 id="Proposed-Guideline-for-Evidence-Based-Categorization-of-Somatic-Variants"><a href="#Proposed-Guideline-for-Evidence-Based-Categorization-of-Somatic-Variants" class="headerlink" title="Proposed Guideline for Evidence-Based Categorization of Somatic Variants"></a>Proposed Guideline for Evidence-Based Categorization of Somatic Variants</h4><p>体细胞变异包括SNV，插入缺失，基因组重排产生的融合基因和CNV。与胚系变异的解释（侧重于特定疾病的变异性或疾病因果关系）不同，体细胞变异的解释应着重于其对临床的影响。如果变异预测对特定疗法的敏感性，耐药性或毒性，改变基因的功能（可以被批准的或研究用的药物靶向），则该变异可以视为影响临床护理的生物标志物，可以作为临床试验的纳入标准，影响疾病的预后，帮助确定癌症的诊断或保证实施监视措施以早期发现癌症。因此，临床影响应包括治疗，预后，诊断和预防措施。给定变异的临床影响应根据当前可获得的证据确定。基于变异分类的证据在临床决策中的重要性，可以对其进行不同的权衡。在文献综述和工作组共识的基础上，我们建议将临床和实验证据分为四个级别 </p>
<ol>
<li>Level A, biomarkers that predict response or resistance to<br>US FDA-approved therapies for a specific type of tumor<br>or have been included in professional guidelines as<br>therapeutic, diagnostic, and/or prognostic biomarkers for<br>specific types of tumors;</li>
<li>Level B, biomarkers that predict response or resistance to<br>a therapy based on well-powered studies with consensus<br>from experts in the field, or have diagnostic and/or<br>prognostic significance of certain diseases based on wellpowered<br>studies with expert consensus;</li>
<li>Level C, biomarkers that predict response or resistance to<br>therapies approved by FDA or professional societies for a<br>different tumor type (ie, off-label use of a drug), serve as<br>inclusion criteria for clinical trials, or have diagnostic<br>and/or prognostic significance based on the results of<br>multiple small studies;</li>
<li>Level D, biomarkers that show plausible therapeutic<br>significance based on preclinical studies, or may assist<br>disease diagnosis and/or prognosis themselves or along<br>with other biomarkers based on small studies or multiple<br>case reports with no consensus.</li>
</ol>
<p><img src="/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/Categories_of_Clinical_andor_Experimental_Evidence.jpg" alt="image"></p>
<p>可以将这些证据水平分配给基因组变异体，以确定其临床影响的重要性。 我们建议根据体细胞疾病的临床影响将其序列变体分为四类：</p>
<ul>
<li>tier I，具有强烈临床意义的变体（A和B级证据）； </li>
<li>tier II，具有潜在临床意义的变体（C或D级证据）； </li>
<li>tier III，临床意义未知的变体； </li>
<li>tier IV，良性或可能良性的变体 </li>
</ul>
<p><img src="/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/体细胞变异等级和Level关系.jpg" alt="image"></p>
<h4 id="各类别判断依据"><a href="#各类别判断依据" class="headerlink" title="各类别判断依据"></a>各类别判断依据</h4><h5 id="Tier-I-Variants-Variants-with-Strong-Clinical-Significance-Level-A-and-B-Evidence"><a href="#Tier-I-Variants-Variants-with-Strong-Clinical-Significance-Level-A-and-B-Evidence" class="headerlink" title="Tier I Variants: Variants with Strong Clinical Significance (Level A and B Evidence)"></a>Tier I Variants: Variants with Strong Clinical Significance (Level A and B Evidence)</h5><p><img src="/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/Tier_I.jpg" alt="image"></p>
<h5 id="Tier-II-Variants-Variants-with-Potential-Clinical-Significance-Level-C-and-D-Evidence"><a href="#Tier-II-Variants-Variants-with-Potential-Clinical-Significance-Level-C-and-D-Evidence" class="headerlink" title="Tier II Variants: Variants with Potential Clinical Significance (Level C and D Evidence)"></a>Tier II Variants: Variants with Potential Clinical Significance (Level C and D Evidence)</h5><p><img src="/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/Tier_II.jpg" alt="image"></p>
<h5 id="Tier-III-Variants-Variants-of-Unknown-Significance"><a href="#Tier-III-Variants-Variants-of-Unknown-Significance" class="headerlink" title="Tier III Variants: Variants of Unknown Significance"></a>Tier III Variants: Variants of Unknown Significance</h5><p><img src="/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/Tier_III.jpg" alt="image"></p>
<h5 id="Tier-IV-Variants-Benign-or-Likely-Benign"><a href="#Tier-IV-Variants-Benign-or-Likely-Benign" class="headerlink" title="Tier IV Variants: Benign or Likely Benign"></a>Tier IV Variants: Benign or Likely Benign</h5><p><img src="/2021/02/19/2021-02-19.Standards-and-Guidelines-for-the-Interpretation-and-Reporting-of-Sequence-Variants-in-Cancer/Tier_IV.jpg" alt="image"></p>
<h5 id="Germline-Variants-Identified-during-Cancer-Testing"><a href="#Germline-Variants-Identified-during-Cancer-Testing" class="headerlink" title="Germline Variants Identified during Cancer Testing"></a>Germline Variants Identified during Cancer Testing</h5><p>在体细胞致癌突变的临床实验室研究中，重要的是将获得的体细胞变异体与遗传的胚系变异体区分开。大多数胚系变异体都是遗传变异，这些变异可以代代相传，并且通常在100％的细胞中具有变异，导致等位基因分数为0.5或1.0。体细胞变异是在出生后获得的，通常是由于DNA复制或修复错误或环境侵害引起的。由于存在污染正常组织的情况，因此即使在表面上纯净的肿瘤样本中，体细胞变异的等位基因分数通常&lt;0.5。实验室必须正确识别可能与诊断，预后，治疗干预和/或临床试验选择有关的体细胞突变，并且不得将高频体细胞突变错误识别为胚系变异，因为这可能具有重大的临床意义。同样，实验室必须认识到可能导致癌症易感综合症的胚系变异，这将对患者和其他家庭成员产生医疗保健影响。</p>
<p>为了帮助对变异进行分类，一些实验室在对正常的，匹配的对照DNA样品进行测序的同时，对肿瘤DNA进行了平行测试。当正常，匹配的对照组织与肿瘤一起测序时，胚系变异通常很明显。<strong> In this case, the laboratory must have policies that address detection, disclosure/nondisclosure, and interpretation/reporting of germline variants (see section below).</strong> 当分析中未包括匹配的对照时，实验室应具有可用于推断变体是体细胞或胚系的标准。指定胚系的主要标准是VAF，对于杂合变体，应为约50％，对于纯合变体，应为100％。某些种系变体（例如大插入缺失）可能会导致正常等位基因的优先扩增（在基于扩增子的测试中）或捕获（在基于捕获的测试中），因为这些等位基因的序列同源性丧失，导致&lt;50％用于种系变体的VAF。当在已知的癌症易感综合征基因（例如TP53或BRCA1）中检测到明显的胚系变异时，有关疾病发作年龄的临床信息（年轻人与致癌基因中遗传的种系突变的较高风险相关） ，肿瘤的侧面性（更可能遗传双侧肿瘤）以及癌症的家族或个人病史可以帮助确定癌症易感性的可能性。文献综述和数据库查询还可以帮助确定该变异体先前是否已被报告为易感综合征患者的复发种系变异体。构成突变的有用数据库包括：</p>
<ul>
<li>Online Mendelian Inheritance in Man (National Center for Biotechnology Information, <a href="http://www.ncbi.nlm.nih.gov/" target="_blank" rel="noopener">http://www.ncbi.nlm.nih.gov/</a> omim, last accessed March 6, 2016), </li>
<li>Human Gene Mutation Database (<a href="http://www.hgmd.cf.ac.uk/ac/index.php" target="_blank" rel="noopener">http://www.hgmd.cf.ac.uk/ac/index.php</a>, last accessed March 6, 2016), </li>
<li>ClinVar</li>
<li>locus-specific databases</li>
</ul>
<p>实验室应制定一项政策，对恶性肿瘤中发现的遗传变异进行测试，以在收到患者的适当同意或根据临床医生的要求后，使用经过临床验证的生殖测试来确认变异的生殖或体细胞来源。 ACMG建议在胚系测试中显示阳性结果，以显示53个基因的阳性结果，其中大约一半与可能在体细胞检测面板上的癌症易感性基因有关。即使在仅将该胚系作为一部分进行评估的情况下，也建议进行公开 肿瘤/正常研究。 通过推断，在仅肿瘤的体细胞突变研究中也考虑种系致病变异的可能性似乎是谨慎的。 </p>
<blockquote>
<p><strong>在单样本无法判断变异是体系还是胚系变异是，添加局限性说明</strong><br>When paired germline samples are not used, NGS analysis<br>does not distinguish germline and somatic variants, and<br>sequencing results may contain both findings. In this case,<br>findings can be reported with a disclaimer that the NGS test<br>used does not allow definitive differentiation between<br>germline and somatic variants. In certain settings, a germline<br>variant may be suspected (eg, MAF 40% to 60%). However,<br>this interpretation should be made with caution and correlated<br>with tumor cellularity. If a germline variant is suspected,<br>testing of a patient germline sample (eg, blood in patients<br>with solid tumors) can be suggested if clinically indicated<br>after an appropriate patient consent. The reports should<br>include a statement addressing the manner in which the<br>distinction between somatic and germline alterations is made,<br>and indications of remaining uncertainty, where appropriate.</p>
</blockquote>
<p>“</p>
<h4 id="Interpretation-and-Reporting"><a href="#Interpretation-and-Reporting" class="headerlink" title="Interpretation and Reporting"></a>Interpretation and Reporting</h4><h5 id="Nomenclature"><a href="#Nomenclature" class="headerlink" title="Nomenclature"></a>Nomenclature</h5><p>所有检测到的遗传变异均应按照HUGO基因命名委员会的指定进行注释和报告:</p>
<ul>
<li>基因名参考   ：HGNC :<a href="http://www.genenames.org" target="_blank" rel="noopener">http://www.genenames.org</a></li>
<li>变异写法参考：HGVS，http：//<a href="http://www.hgvs.org" target="_blank" rel="noopener">www.hgvs.org</a> </li>
</ul>
<p>SNV和插入缺失应使用p.和c. 符号进行报告（例如，BRAFp.V600E，c.1799T&gt; A）。</p>
<p>SV，列出两个融合的基因伴侣之间用斜杠分隔（例如，EML4 / ALK fusion）。</p>
<p>CNV应以表格形式报告为拷贝数GAIN或LOSS。</p>
<p>如果适用，可以包括基因/基因组基因座的基因组坐标。<br>可以在适当的时候报告数字拷贝数的变化[例如，EGFR拷贝数GAIN（拷贝数比25）； CDKN2A拷贝数LOSS]。</p>
<p>使用标准术语不会超过与临床团队进行清晰明确沟通的需要。 根据需要，除了标准术语外，还应包括口语命名法，以便医生阅读报告并使用它们来确定治疗时向医生清晰地传达含义。例如，TERT启动子变体的报告可以为1-124C&gt; T（ HGVS命名法，然后是括号中的口语命名法（TERTC228T）。使用HGVS命名法报告基因组变体以明确将变体重新映射到参考基因组，而口语命名法则向临床团队传达了明确的信息。</p>
<h5 id="Other-Reporting-Elements"><a href="#Other-Reporting-Elements" class="headerlink" title="Other Reporting Elements"></a>Other Reporting Elements</h5><p>除了检测到的变体之外，报告还应包含其他一些元素，这些元素可能与更深入的结果分析或与随时间推移从该患者获得的其他结果进行比较有关，例如基因组座标，基因组构建和转录参考序列（ 例如NM_004333.4），前提是此信息不会损害患者和临床提供者解释该报告中与之直接相关的基本要素的能力。</p>
<p>本节或对结果进行扩展说明的另一节中，远离顶部结果。应评估等位基因分数（VAF）和覆盖率，并在适当时包括在报告中。该报告应包括所用NGS测定的测序覆盖率临界值。<strong>在报告中应声明所有未满足最低要求的测序覆盖率标准的基因和/或热点均已失败。</strong></p>
<p><strong>报告不应仅限于肯定的发现</strong>。 I类药物/癌症组合应包括相关的阴性结果（例如，肺癌患者中明确缺乏EGFR突变或黑色素瘤患者中明确缺乏BRAF突变）。如果存在不确定性， 必须在报告中进行沟通；这包括序列质量，样品充分性，肿瘤含量和生物医学知识的问题。 </p>
<h5 id="Reporting-of-Germline-Variants"><a href="#Reporting-of-Germline-Variants" class="headerlink" title="Reporting of Germline Variants"></a>Reporting of Germline Variants</h5><p>希望对配对的胚系样品进行并发分析，因为这可以澄清解释。但是，这并不总是实用的，因此也不是必需的。当有成对的胚系样品可用时，测序管线可允许从体细胞获得性变体中分离出胚系发现。通常，仅解释和报告体细胞变异。如果患者或临床医生要求对胚系发现进行进一步分析，则可以在以后重新查看胚系NGS数据，并在获得患者适当同意后进行报告。成对的胚系测试可能需要获得同意并提供相关文件，以符合当地法律和政策。如果NGS小组的某些基因中未报告胚系变体，则初始报告应特别说明这一事实。当不使用成对的胚系样品时，NGS分析不能区分胚系和体细胞变体，测序结果可能包含这两个发现。在这种情况下，可以用免责声明报告使用的NGS测试不能明确区分胚系和体细胞变体。在某些情况下，可能会怀疑胚系变异（例如MAF 40％至60％）。然而，这种解释应谨慎进行，并与肿瘤细胞数量有关。如果怀疑胚系变异，如果在适当的患者同意后临床上有指示，可以建议对患者胚系样本（例如，实体瘤患者的血液）进行测试。报告中应包含一份声明，说明在体细胞和胚系变化之间进行区分的方式，并在适当情况下指示仍存在不确定性。首先测试癌症样品和稍后再配对的胚系样品的实验室可以选择进行以下操作： </p>
<ul>
<li>将包含胚系样品发现的结果发布到癌症报告的附录中，</li>
<li>在胚系报告中单独发表一份关于胚系变体的报告，并附上唯一的解释性声明，并在初始癌症报告中添加一份附录。</li>
<li>作为单独的胚系报告和另外的单独报告发布，该报告整合了癌症和胚系样品的结果。</li>
</ul>
<p>生殖细胞变体应按照ACMG / AMP指南进行报告。如果订购了针对癌症易感基因的生殖细胞测试，则生殖细胞变异的报告应遵循ACMG / AMP指南。应提供遗传咨询和转诊给临床医学遗传学家。实验室应制定政策，以报告重要性不明的变体并披露次要发现，包括在何种情况下将报告或不报告此类发现</p>
<h5 id="Reporting-the-Clinical-Significance-of-DetectedVariants"><a href="#Reporting-the-Clinical-Significance-of-DetectedVariants" class="headerlink" title="Reporting the Clinical Significance of DetectedVariants"></a>Reporting the Clinical Significance of DetectedVariants</h5><p>对检测到的遗传变异提供解释性注释是很有用的，该变异将这种变异置于临床病理背景下可为管理决策提供依据。这对于I级和II级突变至关重要。必须对III级变体的详细分析与目标，使报告中最关键的信息保持简洁，清晰并突出显示的目的相平衡。这些评论可能包括该变体对于特定肿瘤类型，对生化途径的影响以及相关癌症的患病率的功能，预后或预测意义。但是，建议应该简短明了，并应谨慎措辞，并应理解治疗或其他患者管理决定是基于除遗传改变以外的许多医学信息，其中许多信息对于分子专业报告是不可用的。重要的是要认识到，治疗的适用性是基于许多因素，而不是根据测试申请书上写的诊断和通过测试发现的基因型。这些因素通常是分子专业报告结果所未知的（即，存在混杂的医学状况，例如葡萄糖耐量不足，自身免疫性疾病或心力衰竭），并且在推荐特定疗法时未考虑这些其他因素会导致混淆，患者与肿瘤科团队之间的冲突以及焦虑。 NGS实验室报告中的治疗建议应以证据为依据，应与患者的癌症诊断相关，并应使用某种语言来明确指出该报告包含结合实验室可用数据点的广义治疗建议（例如，diag-疾病和基因型），但需要将其他因素纳入为每个个体制定治疗计划的过程中。尽管可以接受有关相关试验的一般性陈述或已发表试验的引用结果，但不应针对具体临床试验提出建议。 </p>

      
    </div>
    
    
    

    
      <div>
        <div>
    
        <div style="text-align:center;color: #ccc;font-size:14px;">-------------本文结束<i class="fa fa-paw"></i>感谢您的阅读-------------</div>
    
</div>

      </div>
    
    

    

    

    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="/tags/Standards-and-guidelines/" rel="tag"><i class="fa fa-tag"></i> Standards and guidelines</a>
          
        </div>
      

      
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/2021/02/10/Python-包-python-pptx生成PPT/" rel="next" title="Python-包-python-pptx生成PPT">
                <i class="fa fa-chevron-left"></i> Python-包-python-pptx生成PPT
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/2021/02/19/2021-02-19.Standards-and-guidelines-for-the-interpretation-of-sequence-variants/" rel="prev" title="Standards and guidelines for the interpretation of sequence variants">
                Standards and guidelines for the interpretation of sequence variants <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>



    <div class="post-spread">
      
        <!-- JiaThis Button BEGIN -->
<div class="jiathis_style">
  <a class="jiathis_button_tsina"></a>
  <a class="jiathis_button_tqq"></a>
  <a class="jiathis_button_weixin"></a>
  <a class="jiathis_button_cqq"></a>
  <a class="jiathis_button_douban"></a>
  <a class="jiathis_button_renren"></a>
  <a class="jiathis_button_qzone"></a>
  <a class="jiathis_button_kaixin001"></a>
  <a class="jiathis_button_copy"></a>
  <a href="http://www.jiathis.com/share" class="jiathis jiathis_txt jiathis_separator jtico jtico_jiathis" target="_blank"></a>
  <a class="jiathis_counter_style"></a>
</div>
<script type="text/javascript" >
  var jiathis_config={
    hideMore:false
  }
</script>
<script type="text/javascript" src="http://v3.jiathis.com/code/jia.js" charset="utf-8"></script>
<!-- JiaThis Button END -->

      
    </div>
  </div>


          </div>
          


          
  <div class="comments" id="comments">
    
      <div id="lv-container" data-id="city" data-uid="MTAyMC8zNTI3OS8xMTgxNQ"></div>
    
  </div>


        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap" >
            文章目录
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview">
            站点概览
          </li>
        </ul>
      

      <section class="site-overview sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image"
               src="/uploads/avatar.jpg"
               alt="Ben-air" />
          <p class="site-author-name" itemprop="name">Ben-air</p>
           
              <p class="site-description motion-element" itemprop="description"></p>
          
        </div>
        <nav class="site-state motion-element">

          
            <div class="site-state-item site-state-posts">
              <a href="/archives/">
                <span class="site-state-item-count">116</span>
                <span class="site-state-item-name">日志</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-categories">
              <a href="/categories/index.html">
                <span class="site-state-item-count">41</span>
                <span class="site-state-item-name">分类</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-tags">
              <a href="/tags/index.html">
                <span class="site-state-item-count">57</span>
                <span class="site-state-item-name">标签</span>
              </a>
            </div>
          

        </nav>

        

        <div class="links-of-author motion-element">
          
            
              <span class="links-of-author-item">
                <a href="https://github.com/Ben-unbelieveable" target="_blank" title="GitHub">
                  
                    <i class="fa fa-fw fa-github"></i>
                  
                    
                      GitHub
                    
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="your-twitter-url" target="_blank" title="Twitter">
                  
                    <i class="fa fa-fw fa-twitter"></i>
                  
                    
                      Twitter
                    
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="2432065975" target="_blank" title="Wechat">
                  
                    <i class="fa fa-fw fa-globe"></i>
                  
                    
                      Wechat
                    
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="your-weibo-url" target="_blank" title="Weibo">
                  
                    <i class="fa fa-fw fa-weibo"></i>
                  
                    
                      Weibo
                    
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="your-douban-url" target="_blank" title="DouBan">
                  
                    <i class="fa fa-fw fa-globe"></i>
                  
                    
                      DouBan
                    
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="https://www.zhihu.com/people/dan-mo-39/activities" target="_blank" title="ZhiHu">
                  
                    <i class="fa fa-fw fa-globe"></i>
                  
                    
                      ZhiHu
                    
                </a>
              </span>
            
          
        </div>

        
        

        
        

        


      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-3"><a class="nav-link" href="#参考文献"><span class="nav-number">1.</span> <span class="nav-text">参考文献</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#摘要"><span class="nav-number">2.</span> <span class="nav-text">摘要</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#分类"><span class="nav-number">2.1.</span> <span class="nav-text">分类</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#导读"><span class="nav-number">3.</span> <span class="nav-text">导读</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#对多个公司的报告情况调研结果如下"><span class="nav-number">3.1.</span> <span class="nav-text">对多个公司的报告情况调研结果如下</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#数据库"><span class="nav-number">4.</span> <span class="nav-text">数据库</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Genomic-Databases"><span class="nav-number">4.1.</span> <span class="nav-text">Genomic Databases</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Reference-Sequence-Databases"><span class="nav-number">4.2.</span> <span class="nav-text">Reference Sequence Databases</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Population-Databases"><span class="nav-number">4.3.</span> <span class="nav-text">Population Databases</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Cancer-Specific-Databases"><span class="nav-number">4.4.</span> <span class="nav-text">Cancer-Specific Databases</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Constitutional-Variant-Databases"><span class="nav-number">4.5.</span> <span class="nav-text">Constitutional Variant Databases</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Internal-Laboratory-Generated-Databases"><span class="nav-number">4.6.</span> <span class="nav-text">Internal (Laboratory-Generated) Databases</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#In-Silico-Computational-Prediction-Algorithms"><span class="nav-number">5.</span> <span class="nav-text">In Silico (Computational) Prediction Algorithms</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Variant-Identification-and-Annotation"><span class="nav-number">6.</span> <span class="nav-text">Variant Identification and Annotation</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Proposed-Guideline-for-Evidence-Based-Categorization-of-Somatic-Variants"><span class="nav-number">6.1.</span> <span class="nav-text">Proposed Guideline for Evidence-Based Categorization of Somatic Variants</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#各类别判断依据"><span class="nav-number">6.2.</span> <span class="nav-text">各类别判断依据</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#Tier-I-Variants-Variants-with-Strong-Clinical-Significance-Level-A-and-B-Evidence"><span class="nav-number">6.2.1.</span> <span class="nav-text">Tier I Variants: Variants with Strong Clinical Significance (Level A and B Evidence)</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#Tier-II-Variants-Variants-with-Potential-Clinical-Significance-Level-C-and-D-Evidence"><span class="nav-number">6.2.2.</span> <span class="nav-text">Tier II Variants: Variants with Potential Clinical Significance (Level C and D Evidence)</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#Tier-III-Variants-Variants-of-Unknown-Significance"><span class="nav-number">6.2.3.</span> <span class="nav-text">Tier III Variants: Variants of Unknown Significance</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#Tier-IV-Variants-Benign-or-Likely-Benign"><span class="nav-number">6.2.4.</span> <span class="nav-text">Tier IV Variants: Benign or Likely Benign</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#Germline-Variants-Identified-during-Cancer-Testing"><span class="nav-number">6.2.5.</span> <span class="nav-text">Germline Variants Identified during Cancer Testing</span></a></li></ol></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Interpretation-and-Reporting"><span class="nav-number">6.3.</span> <span class="nav-text">Interpretation and Reporting</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#Nomenclature"><span class="nav-number">6.3.1.</span> <span class="nav-text">Nomenclature</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#Other-Reporting-Elements"><span class="nav-number">6.3.2.</span> <span class="nav-text">Other Reporting Elements</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#Reporting-of-Germline-Variants"><span class="nav-number">6.3.3.</span> <span class="nav-text">Reporting of Germline Variants</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#Reporting-the-Clinical-Significance-of-DetectedVariants"><span class="nav-number">6.3.4.</span> <span class="nav-text">Reporting the Clinical Significance of DetectedVariants</span></a></li></ol></li></ol></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright" >
  
  &copy; 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Ben-air</span>
</div>


        
<div class="busuanzi-count">
  <script async src="https://dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>

  
    <span class="site-uv">
      <i class="fa fa-user"></i>
      <span class="busuanzi-value" id="busuanzi_value_site_uv"></span>
      
    </span>
  

  
    <span class="site-pv">
      <i class="fa fa-eye"></i>
      <span class="busuanzi-value" id="busuanzi_value_site_pv"></span>
      
    </span>
  
</div>


        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  


  











  
  <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>

  
  <script type="text/javascript" src="/lib/canvas-nest/canvas-nest.min.js"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.2"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.2"></script>



  
  


  <script type="text/javascript" src="/js/src/affix.js?v=5.1.2"></script>

  <script type="text/javascript" src="/js/src/schemes/pisces.js?v=5.1.2"></script>



  
  <script type="text/javascript" src="/js/src/scrollspy.js?v=5.1.2"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=5.1.2"></script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.2"></script>



  


  




	





  





  
    <script type="text/javascript">
      (function(d, s) {
        var j, e = d.getElementsByTagName(s)[0];
        if (typeof LivereTower === 'function') { return; }
        j = d.createElement(s);
        j.src = 'https://cdn-city.livere.com/js/embed.dist.js';
        j.async = true;
        e.parentNode.insertBefore(j, e);
      })(document, 'script');
    </script>
  






  

  <script type="text/javascript">
    // Popup Window;
    var isfetched = false;
    var isXml = true;
    // Search DB path;
    var search_path = "search.xml";
    if (search_path.length === 0) {
      search_path = "search.xml";
    } else if (/json$/i.test(search_path)) {
      isXml = false;
    }
    var path = "/" + search_path;
    // monitor main search box;

    var onPopupClose = function (e) {
      $('.popup').hide();
      $('#local-search-input').val('');
      $('.search-result-list').remove();
      $('#no-result').remove();
      $(".local-search-pop-overlay").remove();
      $('body').css('overflow', '');
    }

    function proceedsearch() {
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay"></div>')
        .css('overflow', 'hidden');
      $('.search-popup-overlay').click(onPopupClose);
      $('.popup').toggle();
      var $localSearchInput = $('#local-search-input');
      $localSearchInput.attr("autocapitalize", "none");
      $localSearchInput.attr("autocorrect", "off");
      $localSearchInput.focus();
    }

    // search function;
    var searchFunc = function(path, search_id, content_id) {
      'use strict';

      // start loading animation
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay">' +
          '<div id="search-loading-icon">' +
          '<i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>' +
          '</div>' +
          '</div>')
        .css('overflow', 'hidden');
      $("#search-loading-icon").css('margin', '20% auto 0 auto').css('text-align', 'center');

      $.ajax({
        url: path,
        dataType: isXml ? "xml" : "json",
        async: true,
        success: function(res) {
          // get the contents from search data
          isfetched = true;
          $('.popup').detach().appendTo('.header-inner');
          var datas = isXml ? $("entry", res).map(function() {
            return {
              title: $("title", this).text(),
              content: $("content",this).text(),
              url: $("url" , this).text()
            };
          }).get() : res;
          var input = document.getElementById(search_id);
          var resultContent = document.getElementById(content_id);
          var inputEventFunction = function() {
            var searchText = input.value.trim().toLowerCase();
            var keywords = searchText.split(/[\s\-]+/);
            if (keywords.length > 1) {
              keywords.push(searchText);
            }
            var resultItems = [];
            if (searchText.length > 0) {
              // perform local searching
              datas.forEach(function(data) {
                var isMatch = false;
                var hitCount = 0;
                var searchTextCount = 0;
                var title = data.title.trim();
                var titleInLowerCase = title.toLowerCase();
                var content = data.content.trim().replace(/<[^>]+>/g,"");
                var contentInLowerCase = content.toLowerCase();
                var articleUrl = decodeURIComponent(data.url);
                var indexOfTitle = [];
                var indexOfContent = [];
                // only match articles with not empty titles
                if(title != '') {
                  keywords.forEach(function(keyword) {
                    function getIndexByWord(word, text, caseSensitive) {
                      var wordLen = word.length;
                      if (wordLen === 0) {
                        return [];
                      }
                      var startPosition = 0, position = [], index = [];
                      if (!caseSensitive) {
                        text = text.toLowerCase();
                        word = word.toLowerCase();
                      }
                      while ((position = text.indexOf(word, startPosition)) > -1) {
                        index.push({position: position, word: word});
                        startPosition = position + wordLen;
                      }
                      return index;
                    }

                    indexOfTitle = indexOfTitle.concat(getIndexByWord(keyword, titleInLowerCase, false));
                    indexOfContent = indexOfContent.concat(getIndexByWord(keyword, contentInLowerCase, false));
                  });
                  if (indexOfTitle.length > 0 || indexOfContent.length > 0) {
                    isMatch = true;
                    hitCount = indexOfTitle.length + indexOfContent.length;
                  }
                }

                // show search results

                if (isMatch) {
                  // sort index by position of keyword

                  [indexOfTitle, indexOfContent].forEach(function (index) {
                    index.sort(function (itemLeft, itemRight) {
                      if (itemRight.position !== itemLeft.position) {
                        return itemRight.position - itemLeft.position;
                      } else {
                        return itemLeft.word.length - itemRight.word.length;
                      }
                    });
                  });

                  // merge hits into slices

                  function mergeIntoSlice(text, start, end, index) {
                    var item = index[index.length - 1];
                    var position = item.position;
                    var word = item.word;
                    var hits = [];
                    var searchTextCountInSlice = 0;
                    while (position + word.length <= end && index.length != 0) {
                      if (word === searchText) {
                        searchTextCountInSlice++;
                      }
                      hits.push({position: position, length: word.length});
                      var wordEnd = position + word.length;

                      // move to next position of hit

                      index.pop();
                      while (index.length != 0) {
                        item = index[index.length - 1];
                        position = item.position;
                        word = item.word;
                        if (wordEnd > position) {
                          index.pop();
                        } else {
                          break;
                        }
                      }
                    }
                    searchTextCount += searchTextCountInSlice;
                    return {
                      hits: hits,
                      start: start,
                      end: end,
                      searchTextCount: searchTextCountInSlice
                    };
                  }

                  var slicesOfTitle = [];
                  if (indexOfTitle.length != 0) {
                    slicesOfTitle.push(mergeIntoSlice(title, 0, title.length, indexOfTitle));
                  }

                  var slicesOfContent = [];
                  while (indexOfContent.length != 0) {
                    var item = indexOfContent[indexOfContent.length - 1];
                    var position = item.position;
                    var word = item.word;
                    // cut out 100 characters
                    var start = position - 20;
                    var end = position + 80;
                    if(start < 0){
                      start = 0;
                    }
                    if (end < position + word.length) {
                      end = position + word.length;
                    }
                    if(end > content.length){
                      end = content.length;
                    }
                    slicesOfContent.push(mergeIntoSlice(content, start, end, indexOfContent));
                  }

                  // sort slices in content by search text's count and hits' count

                  slicesOfContent.sort(function (sliceLeft, sliceRight) {
                    if (sliceLeft.searchTextCount !== sliceRight.searchTextCount) {
                      return sliceRight.searchTextCount - sliceLeft.searchTextCount;
                    } else if (sliceLeft.hits.length !== sliceRight.hits.length) {
                      return sliceRight.hits.length - sliceLeft.hits.length;
                    } else {
                      return sliceLeft.start - sliceRight.start;
                    }
                  });

                  // select top N slices in content

                  var upperBound = parseInt('1');
                  if (upperBound >= 0) {
                    slicesOfContent = slicesOfContent.slice(0, upperBound);
                  }

                  // highlight title and content

                  function highlightKeyword(text, slice) {
                    var result = '';
                    var prevEnd = slice.start;
                    slice.hits.forEach(function (hit) {
                      result += text.substring(prevEnd, hit.position);
                      var end = hit.position + hit.length;
                      result += '<b class="search-keyword">' + text.substring(hit.position, end) + '</b>';
                      prevEnd = end;
                    });
                    result += text.substring(prevEnd, slice.end);
                    return result;
                  }

                  var resultItem = '';

                  if (slicesOfTitle.length != 0) {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + highlightKeyword(title, slicesOfTitle[0]) + "</a>";
                  } else {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + title + "</a>";
                  }

                  slicesOfContent.forEach(function (slice) {
                    resultItem += "<a href='" + articleUrl + "'>" +
                      "<p class=\"search-result\">" + highlightKeyword(content, slice) +
                      "...</p>" + "</a>";
                  });

                  resultItem += "</li>";
                  resultItems.push({
                    item: resultItem,
                    searchTextCount: searchTextCount,
                    hitCount: hitCount,
                    id: resultItems.length
                  });
                }
              })
            };
            if (keywords.length === 1 && keywords[0] === "") {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-search fa-5x" /></div>'
            } else if (resultItems.length === 0) {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-frown-o fa-5x" /></div>'
            } else {
              resultItems.sort(function (resultLeft, resultRight) {
                if (resultLeft.searchTextCount !== resultRight.searchTextCount) {
                  return resultRight.searchTextCount - resultLeft.searchTextCount;
                } else if (resultLeft.hitCount !== resultRight.hitCount) {
                  return resultRight.hitCount - resultLeft.hitCount;
                } else {
                  return resultRight.id - resultLeft.id;
                }
              });
              var searchResultList = '<ul class=\"search-result-list\">';
              resultItems.forEach(function (result) {
                searchResultList += result.item;
              })
              searchResultList += "</ul>";
              resultContent.innerHTML = searchResultList;
            }
          }

          if ('auto' === 'auto') {
            input.addEventListener('input', inputEventFunction);
          } else {
            $('.search-icon').click(inputEventFunction);
            input.addEventListener('keypress', function (event) {
              if (event.keyCode === 13) {
                inputEventFunction();
              }
            });
          }

          // remove loading animation
          $(".local-search-pop-overlay").remove();
          $('body').css('overflow', '');

          proceedsearch();
        }
      });
    }

    // handle and trigger popup window;
    $('.popup-trigger').click(function(e) {
      e.stopPropagation();
      if (isfetched === false) {
        searchFunc(path, 'local-search-input', 'local-search-result');
      } else {
        proceedsearch();
      };
    });

    $('.popup-btn-close').click(onPopupClose);
    $('.popup').click(function(e){
      e.stopPropagation();
    });
    $(document).on('keyup', function (event) {
      var shouldDismissSearchPopup = event.which === 27 &&
        $('.search-popup').is(':visible');
      if (shouldDismissSearchPopup) {
        onPopupClose();
      }
    });
  </script>





  

  

  

  
  
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({
        tex2jax: {
          inlineMath: [ ['$','$'], ["\\(","\\)"]  ],
          processEscapes: true,
          skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
        }
      });
    </script>

    <script type="text/x-mathjax-config">
      MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for (i=0; i < all.length; i += 1) {
          all[i].SourceElement().parentNode.className += ' has-jax';
        }
      });
    </script>
    <script type="text/javascript" src="//cdn.bootcss.com/mathjax/2.7.1/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
  


  

  

</body>
</html>
