<!DOCTYPE html>



  


<html class="theme-next pisces use-motion" lang="zh-Hans">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>
<meta name="theme-color" content="#222">



  
  
    
    
  <script src="/lib/pace/pace.min.js?v=1.0.2"></script>
  <link href="/lib/pace/pace-theme-minimal.min.css?v=1.0.2" rel="stylesheet">
  <script>
    (function(){
      if(''){
        if (prompt('请输入文章密码') !== ''){
          alert('密码错误！');
          history.back();
        }
      }
    })();
  </script>







<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css" />







<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css" />

<link href="/css/main.css?v=5.1.2" rel="stylesheet" type="text/css" />


  <meta name="keywords" content="Tobeinsert," />








  <link rel="shortcut icon" type="image/x-icon" href="/images/favicon.ico?v=5.1.2" />






<meta name="description" content="官方资料github: ClinCNVgithub: pre-process ngs-bits ClinCNV: multi-sample germline CNV detection in NGS data 部署应用项目代码拉取1234# CNV 分析项目代码包程度# CNV分析输入数据生成的代码包，用于构建ClinCNV分析的输入文件，视情况判断是否需要git clone https://gi">
<meta name="keywords" content="Tobeinsert">
<meta property="og:type" content="article">
<meta property="og:title" content="Software-CNV检测-ClinCNV">
<meta property="og:url" content="http://ben-air.cn/03.应用-01.software_in_NGS/Software-CNV检测-ClinCNV/index.html">
<meta property="og:site_name" content="Ben-air">
<meta property="og:description" content="官方资料github: ClinCNVgithub: pre-process ngs-bits ClinCNV: multi-sample germline CNV detection in NGS data 部署应用项目代码拉取1234# CNV 分析项目代码包程度# CNV分析输入数据生成的代码包，用于构建ClinCNV分析的输入文件，视情况判断是否需要git clone https://gi">
<meta property="og:locale" content="zh-Hans">
<meta property="og:image" content="http://ben-air.cn/03.应用-01.software_in_NGS/Software-CNV检测-ClinCNV/clusteringSolution.png">
<meta property="og:image" content="http://ben-air.cn/03.应用-01.software_in_NGS/Software-CNV检测-ClinCNV/clinCNV-CNVCalling.png">
<meta property="og:updated_time" content="2025-01-07T01:38:13.187Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Software-CNV检测-ClinCNV">
<meta name="twitter:description" content="官方资料github: ClinCNVgithub: pre-process ngs-bits ClinCNV: multi-sample germline CNV detection in NGS data 部署应用项目代码拉取1234# CNV 分析项目代码包程度# CNV分析输入数据生成的代码包，用于构建ClinCNV分析的输入文件，视情况判断是否需要git clone https://gi">
<meta name="twitter:image" content="http://ben-air.cn/03.应用-01.software_in_NGS/Software-CNV检测-ClinCNV/clusteringSolution.png">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Pisces',
    sidebar: {"position":"left","display":"post","offset":12,"offset_float":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="http://ben-air.cn/03.应用-01.software_in_NGS/Software-CNV检测-ClinCNV/"/>





  <title>Software-CNV检测-ClinCNV | Ben-air</title>
  














</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  
  
    
  

  <div class="container sidebar-position-left page-post-detail ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <div class="custom-logo-site-title">
      <a href="/"  class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">Ben-air</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle"></p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br />
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br />
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br />
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br />
            
            标签
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br />
            
            关于
          </a>
        </li>
      

      
        <li class="menu-item menu-item-search">
          
            <a href="javascript:;" class="popup-trigger">
          
            
              <i class="menu-item-icon fa fa-search fa-fw"></i> <br />
            
            搜索
          </a>
        </li>
      
    </ul>
  

  
    <div class="site-search">
      
  <div class="popup search-popup local-search-popup">
  <div class="local-search-header clearfix">
    <span class="search-icon">
      <i class="fa fa-search"></i>
    </span>
    <span class="popup-btn-close">
      <i class="fa fa-times-circle"></i>
    </span>
    <div class="local-search-input-wrapper">
      <input autocomplete="off"
             placeholder="搜索..." spellcheck="false"
             type="text" id="local-search-input">
    </div>
  </div>
  <div id="local-search-result"></div>
</div>



    </div>
  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://ben-air.cn/03.应用-01.software_in_NGS/Software-CNV检测-ClinCNV/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="Ben-air">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="/uploads/avatar.jpg">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Ben-air">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">Software-CNV检测-ClinCNV</h1>
        

        <div class="post-meta">
	  

          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2024-07-17T14:54:04+08:00">
                2024-07-17
              </time>
            
            

            

            
          </span>

          
            <span class="post-category" >
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">分类于</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Tobeinsert/" itemprop="url" rel="index">
                    <span itemprop="name">Tobeinsert</span>
                  </a>
                </span>

                
                
              
            </span>
          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">
      
      

      
        <h1 id="官方资料"><a href="#官方资料" class="headerlink" title="官方资料"></a>官方资料</h1><p><a href="https://github.com/imgag/ClinCNV" target="_blank" rel="noopener">github: ClinCNV</a><br><a href="https://github.com/imgag/ngs-bits" target="_blank" rel="noopener">github: pre-process ngs-bits</a></p>
<p><a href="https://www.biorxiv.org/content/10.1101/2022.06.10.495642v1.full.pdf" target="_blank" rel="noopener">ClinCNV: multi-sample germline CNV detection in NGS data</a></p>
<h1 id="部署应用"><a href="#部署应用" class="headerlink" title="部署应用"></a>部署应用</h1><h2 id="项目代码拉取"><a href="#项目代码拉取" class="headerlink" title="项目代码拉取"></a>项目代码拉取</h2><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span> CNV 分析项目代码包</span><br><span class="line">程度</span><br><span class="line"><span class="meta">#</span> CNV分析输入数据生成的代码包，用于构建ClinCNV分析的输入文件，视情况判断是否需要</span><br><span class="line">git clone https://github.com/imgag/ngs-bits.git</span><br></pre></td></tr></table></figure>
<h2 id="R环境安装基于conda"><a href="#R环境安装基于conda" class="headerlink" title="R环境安装基于conda"></a>R环境安装基于conda</h2><p>创建R基础环境，github说明是基于 3.2.3 版本。但是3.2.3 版本已经不维护。。<br>中间版本可能存在问题：</p>
<ul>
<li>R=3.6.1<br>“MASS”、”party”、”umap” 编译有问题，无法通过 <code>install.packages</code> 安装。其中part的依赖包 <a href="https://forum.posit.co/t/rstudio-libraries-msm-and-mvtnorm-are-not-installing-on-linux-rh-rstudio-server-2023-06-0-build-4-2-1/168152/4" target="_blank" rel="noopener">mvtnorm安装问题</a><br>安装问题。未检索到直接解决方案，</li>
</ul>
<p>经测试相关包安装可以在目前最新的R编译版本（R=4.4.1）中，可以通过<code>install.packages</code> 完成全部依赖包的自动安装。</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">conda create -nClinCNV -c conda-forge r-base=4.4.1</span><br></pre></td></tr></table></figure>
<h3 id="R包安装"><a href="#R包安装" class="headerlink" title="R包安装"></a>R包安装</h3><p>完成镜像后，进入R环境安装所列的依赖包<br><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 能直接安装的</span></span><br><span class="line">install.packages(<span class="string">"optparse"</span>)</span><br><span class="line">install.packages(<span class="string">"robustbase"</span>)</span><br><span class="line">install.packages(<span class="string">"data.table"</span>)</span><br><span class="line">install.packages(<span class="string">"foreach"</span>)</span><br><span class="line">install.packages(<span class="string">"doParallel"</span>)</span><br><span class="line">install.packages(<span class="string">"mclust"</span>)</span><br><span class="line">install.packages(<span class="string">"R.utils"</span>)</span><br><span class="line">install.packages(<span class="string">"RColorBrewer"</span>)</span><br><span class="line">install.packages(<span class="string">"dbscan"</span>)</span><br><span class="line">install.packages(<span class="string">"MASS"</span>)</span><br><span class="line">install.packages(<span class="string">"party"</span>)</span><br><span class="line">install.packages(<span class="string">"umap"</span>)</span><br><span class="line">install.packages(<span class="string">"Rcpp"</span>) <span class="comment"># Rcpp 安装后可以提高分析速度，不安装也可以运行</span></span><br></pre></td></tr></table></figure></p>
<h3 id="R包安装确认"><a href="#R包安装确认" class="headerlink" title="R包安装确认"></a>R包安装确认</h3><p>测试包安装<br><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 能直接安装的</span></span><br><span class="line"><span class="keyword">library</span>(<span class="string">"optparse"</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">"robustbase"</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">"data.table"</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">"foreach"</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">"doParallel"</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">"mclust"</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">"R.utils"</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">"RColorBrewer"</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">"dbscan"</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">"MASS"</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">"party"</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">"umap"</span>)</span><br><span class="line"><span class="comment">#library("Rcpp")</span></span><br></pre></td></tr></table></figure></p>
<h2 id="demo数据运行"><a href="#demo数据运行" class="headerlink" title="demo数据运行"></a>demo数据运行</h2><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span> 设置本地拉取的ClinCNV项目存储路径</span><br><span class="line">fold=/path/to/ClinCNV-master </span><br><span class="line"><span class="meta">#</span> 创建输出目录</span><br><span class="line">mkdir $fold"/results" </span><br><span class="line"><span class="meta">#</span> 运行分析</span><br><span class="line">Rscript $fold"/clinCNV.R" --bed $fold"/samples/bed_file.bed" --normal $fold"/samples/coverages_normal.cov" --out $fold"/results"</span><br></pre></td></tr></table></figure>
<p>可以看到流程读取两个输入文件 <code>.cov</code> 是一个覆盖度的矩阵文件(包含待分析的所有样本). <code>.bed</code> 进行了GC-含量注释的bed文件 (from 0 to 1, should be in 4th column).</p>
<h3 id="bed-file-bed"><a href="#bed-file-bed" class="headerlink" title="bed_file.bed"></a>bed_file.bed</h3><p>bed文件时在第四列注释了GC含量，第五列标注基因名称信息（第四列是必须的，第五列是可选的）。<code>bed</code>文件中不应该包含文件头，或文件头应该是用 <code>#</code> 注释的。</p>
<table>
<thead>
<tr>
<th>列号</th>
<th>说明</th>
<th>备注</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>染色体</td>
<td>染色体位置，chr前缀</td>
</tr>
<tr>
<td>2</td>
<td>起始坐标</td>
<td>int</td>
</tr>
<tr>
<td>3</td>
<td>结束坐标</td>
<td>int</td>
</tr>
<tr>
<td>4</td>
<td>GC含量</td>
<td>real, from 0 to 1</td>
</tr>
<tr>
<td>5</td>
<td>genes</td>
<td>character, 逗号分隔，可选内容</td>
</tr>
</tbody>
</table>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">chr1    12171   12245   0.4595</span><br><span class="line"># or, annotated with genes,</span><br><span class="line">chr1    12171   12245   0.4595  DDX11L1</span><br></pre></td></tr></table></figure>
<h3 id="coverages-normal-cov"><a href="#coverages-normal-cov" class="headerlink" title="coverages_normal.cov"></a>coverages_normal.cov</h3><p><code>.cov</code> 文件是包含染色体区域信息（前三列），每个样本的对应区域的覆盖度（每个样本一列，建议是区域的平均覆盖度）矩阵文件。首行是文件的头文件。</p>
<table>
<thead>
<tr>
<th>列号</th>
<th>说明</th>
<th>备注</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>染色体</td>
<td>染色体位置，chr前缀</td>
</tr>
<tr>
<td>2</td>
<td>起始坐标</td>
<td>int</td>
</tr>
<tr>
<td>3</td>
<td>结束坐标</td>
<td>int</td>
</tr>
<tr>
<td>4</td>
<td>Sample1</td>
<td>样本在对应区段的平均覆盖深度</td>
</tr>
<tr>
<td>….</td>
<td>….</td>
<td>…..</td>
</tr>
<tr>
<td>3+n</td>
<td>sample-n</td>
<td>样本在对应区段的平均覆盖深度</td>
</tr>
</tbody>
</table>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">chr   start     end             Sam1      Sam2</span><br><span class="line">chr1  11166636  11166864        2374.32   1224.54</span><br></pre></td></tr></table></figure>
<h4 id="文件-cov生成"><a href="#文件-cov生成" class="headerlink" title="文件 *.cov生成"></a>文件 *.cov生成</h4><p>建议使用samtools进行区域覆盖深度的生成，一方面这是 <a href="https://github.com/imgag/ClinCNV" target="_blank" rel="noopener">clinCNV</a> 官方文档中提供的一个计算覆盖度的方式，另一方面是samtools在大多数NGS分析过程中都会用到，所以一般不需要我们进行额外的安装。</p>
<h5 id="基于samtools"><a href="#基于samtools" class="headerlink" title="基于samtools"></a>基于samtools</h5><p>统计深度的命令格式如下：<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">samtools bedcov temp.bed pancancer689__DX1790_daiyufen_23S03853625_23B03853625__Cancer.realign.bam</span><br></pre></td></tr></table></figure></p>
<h5 id="基于ngs-bits-不建议"><a href="#基于ngs-bits-不建议" class="headerlink" title="基于ngs-bits(不建议)"></a>基于ngs-bits(不建议)</h5><p><a href="https://github.com/imgag/ngs-bits" target="_blank" rel="noopener">ngs-bits</a> 是 <a href="https://github.com/imgag/ClinCNV" target="_blank" rel="noopener">clinCNV</a> 官方文档中提供的一个计算覆盖度的方式，但是安装过程发现 <code>ngs-bits</code> 安装的底层依赖和最新版的 <code>clinCNV</code> 已经存在冲突。所以建议更换掉 <code>ngs-bits</code> 软件。</p>
<ul>
<li><p>安装</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">conda create -n ngs-bits  -c bioconda  ngs-bits</span><br></pre></td></tr></table></figure>
</li>
<li><p>使用</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">BedCoverage -bam $bamPath -in $bedPath -min_mapq 3 -out $sampleName&quot;.cov&quot;</span><br></pre></td></tr></table></figure>
</li>
</ul>
<h5 id="mosdepth"><a href="#mosdepth" class="headerlink" title="mosdepth"></a>mosdepth</h5><p><code>mosdepth</code> 是另外一个可以实现相关功能的软件。</p>
<ul>
<li><p>安装</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">conda install -c bioconda mosdepth</span><br></pre></td></tr></table></figure>
</li>
<li><p>使用</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">mosdepth --by temp.bed output_prefix input.bam</span><br></pre></td></tr></table></figure>
</li>
<li><p>结果<br>运行后会生成多个结果，其中regions.bed.gz，就是可以用 <code>ClinCNV</code> 进行CNV分析的格式文件（单样本），对每个样本进行处理后，需要对多个样本的结果进行整合。  </p>
</li>
</ul>
<h1 id="结果说明"><a href="#结果说明" class="headerlink" title="结果说明"></a>结果说明</h1><h2 id="批次统计结果"><a href="#批次统计结果" class="headerlink" title="批次统计结果"></a>批次统计结果</h2><p>批次结果有两部分，1. 统计每个样本的平均深度信息，并计算样本的内部噪音；2.对批次内地所有样本，计算样本间的相关性，完成聚类。</p>
<h3 id="样本聚类情况统计"><a href="#样本聚类情况统计" class="headerlink" title="样本聚类情况统计"></a>样本聚类情况统计</h3><p>测试数据载入后，会进行前期的聚类分组，聚类结果见 <code>$fold&quot;/results/clusterization_of_samples.tsv</code> 会输出聚类后所有样本的cluster分类情况(<code>sample_id，cluster_id</code>)<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">20240827__***********47	0</span><br><span class="line">20240827__***********48	0</span><br><span class="line">20240827__***********49	0</span><br><span class="line">20240827__***********50	0</span><br><span class="line">20240827__***********51	0</span><br><span class="line">20240827__***********52	0</span><br><span class="line">20240827__***********53	0</span><br><span class="line">20240827__***********54	0</span><br><span class="line">20240827__***********56	0</span><br><span class="line">20240827__***********57	0</span><br><span class="line">20240827__***********58	0</span><br><span class="line">20240827__***********59	0</span><br><span class="line">20240827__***********60	0</span><br><span class="line">20240827__***********61	0</span><br></pre></td></tr></table></figure></p>
<p>同时也会生成聚类图<code>$fold&quot;/results/clusteringSolution.png&quot;</code><br><img src="/03.应用-01.software_in_NGS/Software-CNV检测-ClinCNV/clusteringSolution.png" alt="clusteringSolution"></p>
<h3 id="样本覆盖度情况统计"><a href="#样本覆盖度情况统计" class="headerlink" title="样本覆盖度情况统计"></a>样本覆盖度情况统计</h3><p><code>$fold&quot;/results/ontargetNormal.summary.xls&quot;</code> 会统计整批分析样本的平均深度信息以及计算得到的噪音信息(noises &lt;- round(apply(gcNormalisedCov[which(!bedFile[,1] %in% c(“chrX”,”chrY”)),], 2, mad), digits = 2))。<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">namesOfOutputFile       noises  avgDepth</span><br><span class="line">20240827__*********47   0.31    183.01652892562</span><br><span class="line">20240827__*********48   0.33    155.842519685039</span><br><span class="line">20240827__*********49   0.33    140.257462686567</span><br><span class="line">20240827__*********50   0.34    142.707692307692</span><br><span class="line">20240827__*********51   0.33    165.570247933884</span><br><span class="line">20240827__*********52   0.34    151.570247933884</span><br><span class="line">20240827__*********53   0.34    119.438016528926</span><br><span class="line">20240827__*********54   0.33    162.801418439716</span><br></pre></td></tr></table></figure></p>
<h2 id="样本CNV检测结果"><a href="#样本CNV检测结果" class="headerlink" title="样本CNV检测结果"></a>样本CNV检测结果</h2><p>带 <code>cnvs</code> 标记的，都是样本的CNV检测结果对应的信息，其中 <code>tsv</code> 文件为 <code>ClinCNV</code> 输出的CNV详细信息（涵盖了bin的输入，CNV的长度，覆盖的基因等。<br>两个 <code>seg</code> 后缀的文件为IGV的<a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/" target="_blank" rel="noopener">tracks文件</a>，</p>
<ul>
<li>_cov.seg，提供了实际覆盖率的值（覆盖率值是 GC 归一化和中位数归一化，以常染色体中的拷贝数 2 为中心）</li>
<li>_cnvs.seg文件中，拷贝数范围只有 [0,1,2,3,4,5,6] ,所有6拷贝以上的CNV，都按6进行输出。</li>
</ul>
<h3 id="cnvs-tsv"><a href="#cnvs-tsv" class="headerlink" title="**_cnvs.tsv"></a><strong>**</strong>_cnvs.tsv</h3><p>最主要的CNV结果，结果由两部分构成，标签标题行和结果表格。</p>
<h4 id="标签标题行示例如下："><a href="#标签标题行示例如下：" class="headerlink" title="标签标题行示例如下："></a>标签标题行示例如下：</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">##ANALYSISTYPE=CLINCNV_GERMLINE_SINGLE</span><br><span class="line">##ClinCNV version: v1.19.0</span><br><span class="line">##Analysis finished on: 2024-09-25 14:54:39.518772</span><br><span class="line">##gender of sample: M</span><br><span class="line">##number of iterations: 1</span><br><span class="line">##quality used at final iteration: 20</span><br><span class="line">##was it outlier after clustering: FALSE</span><br><span class="line">##fraction of outliers: 0.053</span><br></pre></td></tr></table></figure>
<ul>
<li>前 1 行 是技术性的记录。</li>
<li>前 2 行 是软件版本。</li>
<li>前 3 行 是分析的时间。</li>
<li>第 4 行 显示推断的性别。</li>
<li>第 5 行 自从样本中的 CNV 数量变得可接受以来，质量分数增加迭代了多少次。</li>
<li>第 6 行 最终实际使用的质量值。</li>
<li>第 7 行 显示该样本在批次效应聚类后是否为异常值。</li>
<li>第 8 行 异常值的分数，显示有多少个覆盖数据点的 p 值小于 0.05。<em>ps 它不能真正用于 QC 控制，因为较大的非整倍性会导致高比例的异常值。</em></li>
</ul>
<p>The last line – fraction of outliers – shows how many coverage data points had p-value less than 0.05. It can not really be used for the QC control since having large aneuploidy will lead to high fraction of outliers.<br>最后一行（异常值的分数）显示有多少个覆盖数据点的 p 值小于 0.05。它不能真正用于 QC 控制，因为较大的非整倍性会导致高比例的异常值。</p>
<h4 id="结果表格下："><a href="#结果表格下：" class="headerlink" title="结果表格下："></a>结果表格下：</h4><table>
<thead>
<tr>
<th>#chr</th>
<th>start</th>
<th>end</th>
<th>CN_change</th>
<th>loglikelihood</th>
<th>no_of_regions</th>
<th>length_KB</th>
<th>potential_AF</th>
<th>genes</th>
<th>qvalue</th>
</tr>
</thead>
<tbody>
<tr>
<td>chr1</td>
<td>22315703</td>
<td>22336693</td>
<td>1</td>
<td>231</td>
<td>10</td>
<td>20.990</td>
<td>0.076</td>
<td>-</td>
<td>0.00000</td>
</tr>
<tr>
<td>chr1</td>
<td>17248483</td>
<td>17275039</td>
<td>1</td>
<td>48</td>
<td>18</td>
<td>26.556</td>
<td>0.152</td>
<td>-</td>
<td>0.00419</td>
</tr>
<tr>
<td>chr1</td>
<td>25599018</td>
<td>25655627</td>
<td>1</td>
<td>205</td>
<td>11</td>
<td>56.609</td>
<td>0.127</td>
<td>-</td>
<td>0.00000</td>
</tr>
<tr>
<td>chr1</td>
<td>17060426</td>
<td>17086502</td>
<td>3</td>
<td>18</td>
<td>10</td>
<td>26.076</td>
<td>0.114</td>
<td>-</td>
<td>0.04763</td>
</tr>
<tr>
<td>chr1</td>
<td>110230453</td>
<td>110236367</td>
<td>4</td>
<td>25</td>
<td>9</td>
<td>5.914</td>
<td>0.253</td>
<td>-</td>
<td>0.01483</td>
</tr>
</tbody>
</table>
<ul>
<li>#chr： 染色体;</li>
<li>start： CNV的起始;</li>
<li>end：CNV的终止;</li>
<li>CN_change：检测到的变体的拷贝数;</li>
<li>loglikelihood：与基线（二倍体或男性性染色体的 1 个拷贝）相比，检测到的特定区域的拷贝数的可能性。,对数似然 &gt; 10 意味着支持替代拷贝数的证据强度“非常强”，但是，我们建议您保持更高的值（40 或至少 20），因为基因组覆盖度受到影响诸如短插入缺失等众多事件导致比对问题、批次效应、测序深度、技术假象等，而且基因组相当长，因此看到大的对数似然变化并不罕见。</li>
<li>no_of_regions: 显示变体中包含多少个数据点。较长的变体通常更可信（但并非总是如此，例如，对loglikelihood值较小的长变体可能是误报）。</li>
<li>length_KB：CNV的长度（start和end间的距离）</li>
<li>potential_AF：显示特定区域的覆盖率异常较低/较高的频率（如果变体分别为删除/重复）。</li>
<li>genes：输入的.bed文件包含基因信息时， 会填充 genes 信息。</li>
<li>qvalue： Z-test的p值，高p值，则对应区域可能是 CNV多态性 区域。</li>
</ul>
<h3 id="cnvs-seg"><a href="#cnvs-seg" class="headerlink" title="**_cnvs.seg"></a><strong>**</strong>_cnvs.seg</h3><table>
<thead>
<tr>
<th>ID</th>
<th>chr</th>
<th>start</th>
<th>end</th>
<th></th>
<th>value</th>
</tr>
</thead>
<tbody>
<tr>
<td>20240827__24B07832897</td>
<td>chr1</td>
<td>22315703</td>
<td>22336693</td>
<td>231.01</td>
<td>1</td>
</tr>
<tr>
<td>20240827__24B07832897</td>
<td>chr1</td>
<td>17248483</td>
<td>17275039</td>
<td>47.98</td>
<td>1</td>
</tr>
<tr>
<td>20240827__24B07832897</td>
<td>chr1</td>
<td>25599018</td>
<td>25655627</td>
<td>205.2</td>
<td>1</td>
</tr>
</tbody>
</table>
<h3 id="cov-seg"><a href="#cov-seg" class="headerlink" title="**_cov.seg"></a><strong>**</strong>_cov.seg</h3><p>记录了bed上每个区域的异常值和折算的拷贝数信息。</p>
<table>
<thead>
<tr>
<th>ID</th>
<th>chr</th>
<th>start</th>
<th>end</th>
<th>variance</th>
<th>value</th>
</tr>
</thead>
<tbody>
<tr>
<td>20240827__24B07832897</td>
<td>chr1</td>
<td>65498</td>
<td>65638</td>
<td>0.0934</td>
<td>1.98</td>
</tr>
<tr>
<td>20240827__24B07832897</td>
<td>chr1</td>
<td>69036</td>
<td>70008</td>
<td>0.0604</td>
<td>2.3</td>
</tr>
<tr>
<td>20240827__24B07832897</td>
<td>chr1</td>
<td>367648</td>
<td>368607</td>
<td>0.0836</td>
<td>2.24</td>
</tr>
<tr>
<td>20240827__24B07832897</td>
<td>chr1</td>
<td>564537</td>
<td>564657</td>
<td>0.1381</td>
<td>2.14</td>
</tr>
<tr>
<td>20240827__24B07832897</td>
<td>chr1</td>
<td>621085</td>
<td>622044</td>
<td>0.0839</td>
<td>2.17</td>
</tr>
</tbody>
</table>
<h1 id="软件使用"><a href="#软件使用" class="headerlink" title="软件使用"></a>软件使用</h1><h2 id="针对遗传检测"><a href="#针对遗传检测" class="headerlink" title="针对遗传检测"></a><a href="https://github.com/imgag/ClinCNV/blob/master/doc/germline_CNV_analysis.md" target="_blank" rel="noopener">针对遗传检测</a></h2><h3 id="基础分析参数"><a href="#基础分析参数" class="headerlink" title="基础分析参数"></a>基础分析参数</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">Rscript /ifstj2/B2C_RD_H1/Personal/liubo/3.Software/ClinCNV-master/clinCNV.R \</span><br><span class="line"><span class="meta">#</span> 待检测的目标区域文件 bed，详见demo测试部分具体说明</span><br><span class="line">--bed /jdfstj6/B2C_RD/liubo4/product/Large_CNV/00.bed.info/WESv5withSafeRisk.addGCRate.bed \</span><br><span class="line"><span class="meta">#</span> 待检测的目标区域覆盖度文件，详见demo测试部分具体说明</span><br><span class="line">--normal /jdfstj6/B2C_RD/liubo4/product/Large_CNV/by_ClinCNV/20240915_DNBSEQ-T7A2302409150004.mergeCov.tsv \</span><br><span class="line"><span class="meta">#</span> 指定输出目录，将在输出文件夹中创建normal文件夹，结果将放入子文件夹中，每个子文件夹一个样本。</span><br><span class="line">--out /jdfstj6/B2C_RD/liubo4/product/Large_CNV/by_ClinCNV/20240915_DNBSEQ-T7A2302409150004_ClinCNV</span><br></pre></td></tr></table></figure>
<h3 id="可选参数"><a href="#可选参数" class="headerlink" title="可选参数"></a>可选参数</h3><ul>
<li><p>–normalOfftarget &amp; –bedOfftarget<br>添加脱靶覆盖率（不会大幅提高灵敏度，因为种系 CNV 很少与脱靶窗口一样长，但<strong>可以提高特异性，有效去除彼此远离的探针形成的 CNV</strong>，必须提取足够大的脱靶覆盖率样本数，但不一定是normal.cov文件中的所有样本）。<br>e.g.: <code>--normalOfftarget normalOff.cov --bedOfftarget bedFileOff.bed</code></p>
</li>
<li><p>–scoreG<br>平衡灵敏度和特异性（增加阈值–scoreG 会导致更高的特异性和更低的灵敏度，反之亦然，默认值为 20）。<br>e.g.: <code>--scoreG 60</code></p>
</li>
<li><p>–lengthG<br>增加或减少检测到的变体的最小长度（默认 = 2 个<strong>数据点</strong>或更大，实际数据点数量实际上是您在此处指定的 +1，0 表示至少 1 个数据点）。<br>e.g.: <code>--lengthG 0</code></p>
</li>
<li><p>–maxNumGermCNVs &amp; –maxNumIter<br>增加您期望从该样本中获得的 CNV 数量（默认 = 10000，但对于 WGS 样本，最好将其设置为 1000 甚至更大，当超过此阈值时，质量阈值会增加，并且样本会重新分析多次（由 <strong>–maxNumIter</strong> 指定）<br>e.g.: <code>--maxNumGermCNVs 2000 --maxNumIter 5</code></p>
</li>
<li><p>–minimumNumOfElemsInCluster 50<br>将大队列分成几个相似样本的集群，以进行更准确的参数估计并加快工具的速度。您指定集群的最小大小（我们建议将其保持在 20 到 100 之间）：<br>e.g.: <code>--minimumNumOfElemsInCluster 50</code></p>
</li>
<li><p>–numberOfThreads 4<br>通过使用多个核心来加速分析过程中的<strong>某些</strong>部分，以提高分析速度。<br>e.g.: <code>--numberOfThreads 4</code></p>
</li>
</ul>
<h3 id="Polymorphic-CNVs"><a href="#Polymorphic-CNVs" class="headerlink" title="Polymorphic CNVs"></a>Polymorphic CNVs</h3><p>人群频率大于 2.5% 的 CNV 通常更难以使用常规方法检测。因此，我们使用高斯混合模型检测它们。这是同一脚本中的单独例程。要调用此类变体，您需要指定</p>
<ul>
<li>–polymorphicCalling YES<br>如果提供的参数不是 <strong>YES</strong> 而是带有多态区域坐标的.bed文件路径， ClinCNV将在调用过程中忽略他们。</li>
</ul>
<h3 id="嵌合体"><a href="#嵌合体" class="headerlink" title="嵌合体"></a>嵌合体</h3><p>嵌合体CNV的检测范围为 1.1 到 2.9，步长为 0.05。</p>
<ul>
<li>–mosaicism</li>
</ul>
<h1 id="软件原理介绍"><a href="#软件原理介绍" class="headerlink" title="软件原理介绍"></a>软件原理介绍</h1><p>因为目前预期用途是进行胚系的WES应用评估，所以相关方法说明参考的 2022年的<a href="https://www.biorxiv.org/content/10.1101/2022.06.10.495642v1.full.pdf" target="_blank" rel="noopener">ClinCNV: multi-sample germline CNV detection in NGS data</a><br>方法检测整体是基于深度的，算法可以比较容易的将等位基因频率纳入考量，但是文章认为等位基因频率对体系CNV的帮助比较大，但是对于胚系CNV检测作用并没有体系那么大。</p>
<h2 id="method"><a href="#method" class="headerlink" title="method"></a>method</h2><p>检测方法整体分为两部分：</p>
<ol>
<li>数据的标准化；</li>
<li>CNV的Calling（这部分基于作者2019年发布的前一篇文章<a href="https://www.biorxiv.org/content/biorxiv/early/2019/11/11/837971.full.pdf" target="_blank" rel="noopener">ClinCNV: novel method for allele-specific somatic copy-number alterations detection</a>。</li>
</ol>
<h3 id="标准化"><a href="#标准化" class="headerlink" title="标准化"></a>标准化</h3><p>在这边文章所述方法中，作者针对标准化进行了调整，来实现以下目的（源于<a href="https://www.biorxiv.org/content/biorxiv/early/2022/06/13/2022.06.10.495642/DC1/embed/media-1.pdf?download=true" target="_blank" rel="noopener">补充材料</a>）：</p>
<ol>
<li>构建bin文件，分析的区段单元，通常是120bp，可以是WES富集捕获的区段，也可以是其他有依据的划分方式。（前期准备）</li>
<li>针对每个bin文件，统计reads的数目。（前期准备）</li>
<li>利用平方根转换$f(x)=\sqrt(x)$，根据(GC-content, region-length based, variance stabilization)进行方差稳定的数据标准化。</li>
</ol>
<h3 id="CNV检测"><a href="#CNV检测" class="headerlink" title="CNV检测"></a>CNV检测</h3><ol>
<li>在所有样本中，寻找和目标样本的覆盖度相似度最高的样本，根据coverage 对样本进行聚类。</li>
<li>进行样本间的标准化，对聚类到一起的样本使用覆盖度的中位值进行标准化。</li>
<li>对统计模型进行参数估计用来描述不同的拷贝数状态（针对每个样本的每个区域）。</li>
<li>构建似然矩阵，矩阵的行对应预先定义的拷贝数状态，矩阵的列对应我们在 step1 中定义的bin的数目。</li>
<li>对bin进行片段化，并进行结果检出</li>
<li>使用各种QC矩阵进行注释，并对结果进行可视化。<br><img src="/03.应用-01.software_in_NGS/Software-CNV检测-ClinCNV/clinCNV-CNVCalling.png" alt="alt text"><br>引自<a href="https://www.biorxiv.org/content/biorxiv/early/2019/11/11/837971.full.pdf" target="_blank" rel="noopener">2019年发表的第一篇文章</a></li>
</ol>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">graph TB;</span><br><span class="line">A(*bed 提前根据探针或其他方式确定分析的bin)--&gt;C</span><br><span class="line">B(*bam 经过上游预处理得到的样本数据)--&gt;C</span><br><span class="line">C(*cov 针对每个bin统计reads数目)--平方根转换--&gt;D</span><br><span class="line">D(标准化后的覆盖度数据)--样本间覆盖相似度比较--&gt;E</span><br><span class="line">E(样本覆盖度的聚类结果)--聚类一组的样本进行中位值标准化--&gt;F</span><br><span class="line">F(对标准化的覆盖度使用参数模型来描述拷贝数状态)--&gt;G</span><br><span class="line">G(构建似然矩阵)--&gt;H</span><br><span class="line">H(对bin进行片段化生成大区段CNV)--&gt;I</span><br><span class="line">I(生成QC，并进行可视化)</span><br></pre></td></tr></table></figure>

      
    </div>
    
    
    

    
      <div>
        <div>
    
        <div style="text-align:center;color: #ccc;font-size:14px;">-------------本文结束<i class="fa fa-paw"></i>感谢您的阅读-------------</div>
    
</div>

      </div>
    
    

    

    

    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="/tags/Tobeinsert/" rel="tag"><i class="fa fa-tag"></i> Tobeinsert</a>
          
        </div>
      

      
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/03.应用-99.other/Dag-有向无环图/" rel="next" title="Dag-有向无环图">
                <i class="fa fa-chevron-left"></i> Dag-有向无环图
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/01.知识-02.Math/数据标准化-平方根转换/" rel="prev" title="数据标准化-平方根转换">
                数据标准化-平方根转换 <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>



    <div class="post-spread">
      
        <!-- JiaThis Button BEGIN -->
<div class="jiathis_style">
  <a class="jiathis_button_tsina"></a>
  <a class="jiathis_button_tqq"></a>
  <a class="jiathis_button_weixin"></a>
  <a class="jiathis_button_cqq"></a>
  <a class="jiathis_button_douban"></a>
  <a class="jiathis_button_renren"></a>
  <a class="jiathis_button_qzone"></a>
  <a class="jiathis_button_kaixin001"></a>
  <a class="jiathis_button_copy"></a>
  <a href="http://www.jiathis.com/share" class="jiathis jiathis_txt jiathis_separator jtico jtico_jiathis" target="_blank"></a>
  <a class="jiathis_counter_style"></a>
</div>
<script type="text/javascript" >
  var jiathis_config={
    hideMore:false
  }
</script>
<script type="text/javascript" src="http://v3.jiathis.com/code/jia.js" charset="utf-8"></script>
<!-- JiaThis Button END -->

      
    </div>
  </div>


          </div>
          


          
  <div class="comments" id="comments">
    
      <div id="lv-container" data-id="city" data-uid="MTAyMC8zNTI3OS8xMTgxNQ"></div>
    
  </div>


        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap" >
            文章目录
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview">
            站点概览
          </li>
        </ul>
      

      <section class="site-overview sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image"
               src="/uploads/avatar.jpg"
               alt="Ben-air" />
          <p class="site-author-name" itemprop="name">Ben-air</p>
           
              <p class="site-description motion-element" itemprop="description"></p>
          
        </div>
        <nav class="site-state motion-element">

          
            <div class="site-state-item site-state-posts">
              <a href="/archives/">
                <span class="site-state-item-count">211</span>
                <span class="site-state-item-name">日志</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-categories">
              <a href="/categories/index.html">
                <span class="site-state-item-count">69</span>
                <span class="site-state-item-name">分类</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-tags">
              <a href="/tags/index.html">
                <span class="site-state-item-count">93</span>
                <span class="site-state-item-name">标签</span>
              </a>
            </div>
          

        </nav>

        

        <div class="links-of-author motion-element">
          
            
              <span class="links-of-author-item">
                <a href="https://github.com/Ben-unbelieveable" target="_blank" title="GitHub">
                  
                    <i class="fa fa-fw fa-github"></i>
                  
                    
                      GitHub
                    
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="your-twitter-url" target="_blank" title="Twitter">
                  
                    <i class="fa fa-fw fa-twitter"></i>
                  
                    
                      Twitter
                    
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="2432065975" target="_blank" title="Wechat">
                  
                    <i class="fa fa-fw fa-globe"></i>
                  
                    
                      Wechat
                    
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="your-weibo-url" target="_blank" title="Weibo">
                  
                    <i class="fa fa-fw fa-weibo"></i>
                  
                    
                      Weibo
                    
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="your-douban-url" target="_blank" title="DouBan">
                  
                    <i class="fa fa-fw fa-globe"></i>
                  
                    
                      DouBan
                    
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="https://www.zhihu.com/people/dan-mo-39/activities" target="_blank" title="ZhiHu">
                  
                    <i class="fa fa-fw fa-globe"></i>
                  
                    
                      ZhiHu
                    
                </a>
              </span>
            
          
        </div>

        
        

        
        

        


      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#官方资料"><span class="nav-number">1.</span> <span class="nav-text">官方资料</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#部署应用"><span class="nav-number">2.</span> <span class="nav-text">部署应用</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#项目代码拉取"><span class="nav-number">2.1.</span> <span class="nav-text">项目代码拉取</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#R环境安装基于conda"><span class="nav-number">2.2.</span> <span class="nav-text">R环境安装基于conda</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#R包安装"><span class="nav-number">2.2.1.</span> <span class="nav-text">R包安装</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#R包安装确认"><span class="nav-number">2.2.2.</span> <span class="nav-text">R包安装确认</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#demo数据运行"><span class="nav-number">2.3.</span> <span class="nav-text">demo数据运行</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#bed-file-bed"><span class="nav-number">2.3.1.</span> <span class="nav-text">bed_file.bed</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#coverages-normal-cov"><span class="nav-number">2.3.2.</span> <span class="nav-text">coverages_normal.cov</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#文件-cov生成"><span class="nav-number">2.3.2.1.</span> <span class="nav-text">文件 *.cov生成</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#基于samtools"><span class="nav-number">2.3.2.1.1.</span> <span class="nav-text">基于samtools</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#基于ngs-bits-不建议"><span class="nav-number">2.3.2.1.2.</span> <span class="nav-text">基于ngs-bits(不建议)</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#mosdepth"><span class="nav-number">2.3.2.1.3.</span> <span class="nav-text">mosdepth</span></a></li></ol></li></ol></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#结果说明"><span class="nav-number">3.</span> <span class="nav-text">结果说明</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#批次统计结果"><span class="nav-number">3.1.</span> <span class="nav-text">批次统计结果</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#样本聚类情况统计"><span class="nav-number">3.1.1.</span> <span class="nav-text">样本聚类情况统计</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#样本覆盖度情况统计"><span class="nav-number">3.1.2.</span> <span class="nav-text">样本覆盖度情况统计</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#样本CNV检测结果"><span class="nav-number">3.2.</span> <span class="nav-text">样本CNV检测结果</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#cnvs-tsv"><span class="nav-number">3.2.1.</span> <span class="nav-text">**_cnvs.tsv</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#标签标题行示例如下："><span class="nav-number">3.2.1.1.</span> <span class="nav-text">标签标题行示例如下：</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#结果表格下："><span class="nav-number">3.2.1.2.</span> <span class="nav-text">结果表格下：</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#cnvs-seg"><span class="nav-number">3.2.2.</span> <span class="nav-text">**_cnvs.seg</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#cov-seg"><span class="nav-number">3.2.3.</span> <span class="nav-text">**_cov.seg</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#软件使用"><span class="nav-number">4.</span> <span class="nav-text">软件使用</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#针对遗传检测"><span class="nav-number">4.1.</span> <span class="nav-text">针对遗传检测</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#基础分析参数"><span class="nav-number">4.1.1.</span> <span class="nav-text">基础分析参数</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#可选参数"><span class="nav-number">4.1.2.</span> <span class="nav-text">可选参数</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Polymorphic-CNVs"><span class="nav-number">4.1.3.</span> <span class="nav-text">Polymorphic CNVs</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#嵌合体"><span class="nav-number">4.1.4.</span> <span class="nav-text">嵌合体</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#软件原理介绍"><span class="nav-number">5.</span> <span class="nav-text">软件原理介绍</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#method"><span class="nav-number">5.1.</span> <span class="nav-text">method</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#标准化"><span class="nav-number">5.1.1.</span> <span class="nav-text">标准化</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#CNV检测"><span class="nav-number">5.1.2.</span> <span class="nav-text">CNV检测</span></a></li></ol></li></ol></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright" >
  
  &copy; 
  <span itemprop="copyrightYear">2025</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Ben-air</span>
</div>


        
<div class="busuanzi-count">
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>

  
    <span class="post-meta-item-icon">
      <i class="fa fa-user"></i>
    </span>
      已有<span class="busuanzi-value" id="busuanzi_value_site_uv"></span>人访问
  

  
    <span class="post-meta-divider">|</span>
  

  
    <span class="post-meta-item-icon">
      <i class="fa fa-eye"></i>
    </span>
      总访问<span class="busuanzi-value" id="busuanzi_value_site_pv"></span>次
  
</div>


        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  


  











  
  <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>

  
  <script type="text/javascript" src="/lib/canvas-nest/canvas-nest.min.js"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.2"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.2"></script>



  
  


  <script type="text/javascript" src="/js/src/affix.js?v=5.1.2"></script>

  <script type="text/javascript" src="/js/src/schemes/pisces.js?v=5.1.2"></script>



  
  <script type="text/javascript" src="/js/src/scrollspy.js?v=5.1.2"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=5.1.2"></script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.2"></script>



  


  




	





  





  
    <script type="text/javascript">
      (function(d, s) {
        var j, e = d.getElementsByTagName(s)[0];
        if (typeof LivereTower === 'function') { return; }
        j = d.createElement(s);
        j.src = 'https://cdn-city.livere.com/js/embed.dist.js';
        j.async = true;
        e.parentNode.insertBefore(j, e);
      })(document, 'script');
    </script>
  






  

  <script type="text/javascript">
    // Popup Window;
    var isfetched = false;
    var isXml = true;
    // Search DB path;
    var search_path = "search.xml";
    if (search_path.length === 0) {
      search_path = "search.xml";
    } else if (/json$/i.test(search_path)) {
      isXml = false;
    }
    var path = "/" + search_path;
    // monitor main search box;

    var onPopupClose = function (e) {
      $('.popup').hide();
      $('#local-search-input').val('');
      $('.search-result-list').remove();
      $('#no-result').remove();
      $(".local-search-pop-overlay").remove();
      $('body').css('overflow', '');
    }

    function proceedsearch() {
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay"></div>')
        .css('overflow', 'hidden');
      $('.search-popup-overlay').click(onPopupClose);
      $('.popup').toggle();
      var $localSearchInput = $('#local-search-input');
      $localSearchInput.attr("autocapitalize", "none");
      $localSearchInput.attr("autocorrect", "off");
      $localSearchInput.focus();
    }

    // search function;
    var searchFunc = function(path, search_id, content_id) {
      'use strict';

      // start loading animation
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay">' +
          '<div id="search-loading-icon">' +
          '<i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>' +
          '</div>' +
          '</div>')
        .css('overflow', 'hidden');
      $("#search-loading-icon").css('margin', '20% auto 0 auto').css('text-align', 'center');

      $.ajax({
        url: path,
        dataType: isXml ? "xml" : "json",
        async: true,
        success: function(res) {
          // get the contents from search data
          isfetched = true;
          $('.popup').detach().appendTo('.header-inner');
          var datas = isXml ? $("entry", res).map(function() {
            return {
              title: $("title", this).text(),
              content: $("content",this).text(),
              url: $("url" , this).text()
            };
          }).get() : res;
          var input = document.getElementById(search_id);
          var resultContent = document.getElementById(content_id);
          var inputEventFunction = function() {
            var searchText = input.value.trim().toLowerCase();
            var keywords = searchText.split(/[\s\-]+/);
            if (keywords.length > 1) {
              keywords.push(searchText);
            }
            var resultItems = [];
            if (searchText.length > 0) {
              // perform local searching
              datas.forEach(function(data) {
                var isMatch = false;
                var hitCount = 0;
                var searchTextCount = 0;
                var title = data.title.trim();
                var titleInLowerCase = title.toLowerCase();
                var content = data.content.trim().replace(/<[^>]+>/g,"");
                var contentInLowerCase = content.toLowerCase();
                var articleUrl = decodeURIComponent(data.url);
                var indexOfTitle = [];
                var indexOfContent = [];
                // only match articles with not empty titles
                if(title != '') {
                  keywords.forEach(function(keyword) {
                    function getIndexByWord(word, text, caseSensitive) {
                      var wordLen = word.length;
                      if (wordLen === 0) {
                        return [];
                      }
                      var startPosition = 0, position = [], index = [];
                      if (!caseSensitive) {
                        text = text.toLowerCase();
                        word = word.toLowerCase();
                      }
                      while ((position = text.indexOf(word, startPosition)) > -1) {
                        index.push({position: position, word: word});
                        startPosition = position + wordLen;
                      }
                      return index;
                    }

                    indexOfTitle = indexOfTitle.concat(getIndexByWord(keyword, titleInLowerCase, false));
                    indexOfContent = indexOfContent.concat(getIndexByWord(keyword, contentInLowerCase, false));
                  });
                  if (indexOfTitle.length > 0 || indexOfContent.length > 0) {
                    isMatch = true;
                    hitCount = indexOfTitle.length + indexOfContent.length;
                  }
                }

                // show search results

                if (isMatch) {
                  // sort index by position of keyword

                  [indexOfTitle, indexOfContent].forEach(function (index) {
                    index.sort(function (itemLeft, itemRight) {
                      if (itemRight.position !== itemLeft.position) {
                        return itemRight.position - itemLeft.position;
                      } else {
                        return itemLeft.word.length - itemRight.word.length;
                      }
                    });
                  });

                  // merge hits into slices

                  function mergeIntoSlice(text, start, end, index) {
                    var item = index[index.length - 1];
                    var position = item.position;
                    var word = item.word;
                    var hits = [];
                    var searchTextCountInSlice = 0;
                    while (position + word.length <= end && index.length != 0) {
                      if (word === searchText) {
                        searchTextCountInSlice++;
                      }
                      hits.push({position: position, length: word.length});
                      var wordEnd = position + word.length;

                      // move to next position of hit

                      index.pop();
                      while (index.length != 0) {
                        item = index[index.length - 1];
                        position = item.position;
                        word = item.word;
                        if (wordEnd > position) {
                          index.pop();
                        } else {
                          break;
                        }
                      }
                    }
                    searchTextCount += searchTextCountInSlice;
                    return {
                      hits: hits,
                      start: start,
                      end: end,
                      searchTextCount: searchTextCountInSlice
                    };
                  }

                  var slicesOfTitle = [];
                  if (indexOfTitle.length != 0) {
                    slicesOfTitle.push(mergeIntoSlice(title, 0, title.length, indexOfTitle));
                  }

                  var slicesOfContent = [];
                  while (indexOfContent.length != 0) {
                    var item = indexOfContent[indexOfContent.length - 1];
                    var position = item.position;
                    var word = item.word;
                    // cut out 100 characters
                    var start = position - 20;
                    var end = position + 80;
                    if(start < 0){
                      start = 0;
                    }
                    if (end < position + word.length) {
                      end = position + word.length;
                    }
                    if(end > content.length){
                      end = content.length;
                    }
                    slicesOfContent.push(mergeIntoSlice(content, start, end, indexOfContent));
                  }

                  // sort slices in content by search text's count and hits' count

                  slicesOfContent.sort(function (sliceLeft, sliceRight) {
                    if (sliceLeft.searchTextCount !== sliceRight.searchTextCount) {
                      return sliceRight.searchTextCount - sliceLeft.searchTextCount;
                    } else if (sliceLeft.hits.length !== sliceRight.hits.length) {
                      return sliceRight.hits.length - sliceLeft.hits.length;
                    } else {
                      return sliceLeft.start - sliceRight.start;
                    }
                  });

                  // select top N slices in content

                  var upperBound = parseInt('1');
                  if (upperBound >= 0) {
                    slicesOfContent = slicesOfContent.slice(0, upperBound);
                  }

                  // highlight title and content

                  function highlightKeyword(text, slice) {
                    var result = '';
                    var prevEnd = slice.start;
                    slice.hits.forEach(function (hit) {
                      result += text.substring(prevEnd, hit.position);
                      var end = hit.position + hit.length;
                      result += '<b class="search-keyword">' + text.substring(hit.position, end) + '</b>';
                      prevEnd = end;
                    });
                    result += text.substring(prevEnd, slice.end);
                    return result;
                  }

                  var resultItem = '';

                  if (slicesOfTitle.length != 0) {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + highlightKeyword(title, slicesOfTitle[0]) + "</a>";
                  } else {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + title + "</a>";
                  }

                  slicesOfContent.forEach(function (slice) {
                    resultItem += "<a href='" + articleUrl + "'>" +
                      "<p class=\"search-result\">" + highlightKeyword(content, slice) +
                      "...</p>" + "</a>";
                  });

                  resultItem += "</li>";
                  resultItems.push({
                    item: resultItem,
                    searchTextCount: searchTextCount,
                    hitCount: hitCount,
                    id: resultItems.length
                  });
                }
              })
            };
            if (keywords.length === 1 && keywords[0] === "") {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-search fa-5x" /></div>'
            } else if (resultItems.length === 0) {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-frown-o fa-5x" /></div>'
            } else {
              resultItems.sort(function (resultLeft, resultRight) {
                if (resultLeft.searchTextCount !== resultRight.searchTextCount) {
                  return resultRight.searchTextCount - resultLeft.searchTextCount;
                } else if (resultLeft.hitCount !== resultRight.hitCount) {
                  return resultRight.hitCount - resultLeft.hitCount;
                } else {
                  return resultRight.id - resultLeft.id;
                }
              });
              var searchResultList = '<ul class=\"search-result-list\">';
              resultItems.forEach(function (result) {
                searchResultList += result.item;
              })
              searchResultList += "</ul>";
              resultContent.innerHTML = searchResultList;
            }
          }

          if ('auto' === 'auto') {
            input.addEventListener('input', inputEventFunction);
          } else {
            $('.search-icon').click(inputEventFunction);
            input.addEventListener('keypress', function (event) {
              if (event.keyCode === 13) {
                inputEventFunction();
              }
            });
          }

          // remove loading animation
          $(".local-search-pop-overlay").remove();
          $('body').css('overflow', '');

          proceedsearch();
        }
      });
    }

    // handle and trigger popup window;
    $('.popup-trigger').click(function(e) {
      e.stopPropagation();
      if (isfetched === false) {
        searchFunc(path, 'local-search-input', 'local-search-result');
      } else {
        proceedsearch();
      };
    });

    $('.popup-btn-close').click(onPopupClose);
    $('.popup').click(function(e){
      e.stopPropagation();
    });
    $(document).on('keyup', function (event) {
      var shouldDismissSearchPopup = event.which === 27 &&
        $('.search-popup').is(':visible');
      if (shouldDismissSearchPopup) {
        onPopupClose();
      }
    });
  </script>





  

  

  

  
  
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({
        tex2jax: {
          inlineMath: [ ['$','$'], ["\\(","\\)"]  ],
          processEscapes: true,
          skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
        }
      });
    </script>

    <script type="text/x-mathjax-config">
      MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for (i=0; i < all.length; i += 1) {
          all[i].SourceElement().parentNode.className += ' has-jax';
        }
      });
    </script>
    <script type="text/javascript" src="//cdn.bootcss.com/mathjax/2.7.1/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
  


  

  

</body>
</html>
